<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="es" xml:lang="es"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="OMICs Analysis">

<title>Metabarcoding: Tomate 16S usando Qiime2</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="index_files/libs/clipboard/clipboard.min.js"></script>
<script src="index_files/libs/quarto-html/quarto.js"></script>
<script src="index_files/libs/quarto-html/popper.min.js"></script>
<script src="index_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="index_files/libs/quarto-html/anchor.min.js"></script>
<link href="index_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="index_files/libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="index_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="index_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="index_files/libs/bootstrap/bootstrap-a71a25f19e128005acb9c00d38d9c2cf.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<style>html{ scroll-behavior: smooth; }</style>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contenido</h2>
   
  <ul>
  <li><a href="#introducción" id="toc-introducción" class="nav-link active" data-scroll-target="#introducción">Introducción</a></li>
  <li><a href="#importar-datos-pareados-con-qiime2" id="toc-importar-datos-pareados-con-qiime2" class="nav-link" data-scroll-target="#importar-datos-pareados-con-qiime2">Importar datos pareados con Qiime2</a>
  <ul class="collapse">
  <li><a href="#cutadapt" id="toc-cutadapt" class="nav-link" data-scroll-target="#cutadapt">Cutadapt</a>
  <ul class="collapse">
  <li><a href="#explicación-del-comando" id="toc-explicación-del-comando" class="nav-link" data-scroll-target="#explicación-del-comando">Explicación del comando:</a></li>
  <li><a href="#explorar-los-archivos" id="toc-explorar-los-archivos" class="nav-link" data-scroll-target="#explorar-los-archivos">Explorar los archivos</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#filtrado-de-lecturas-con-dada2" id="toc-filtrado-de-lecturas-con-dada2" class="nav-link" data-scroll-target="#filtrado-de-lecturas-con-dada2">Filtrado de lecturas con DADA2</a></li>
  <li><a href="#medidas-de-diversidad" id="toc-medidas-de-diversidad" class="nav-link" data-scroll-target="#medidas-de-diversidad">Medidas de diversidad</a>
  <ul class="collapse">
  <li><a href="#generación-de-un-árbol-filogenético" id="toc-generación-de-un-árbol-filogenético" class="nav-link" data-scroll-target="#generación-de-un-árbol-filogenético">Generación de un árbol filogenético</a></li>
  <li><a href="#métricas-general-de-filogenética" id="toc-métricas-general-de-filogenética" class="nav-link" data-scroll-target="#métricas-general-de-filogenética">Métricas general de filogenética</a></li>
  <li><a href="#alfa-diversidad-α-diversidad" id="toc-alfa-diversidad-α-diversidad" class="nav-link" data-scroll-target="#alfa-diversidad-α-diversidad">Alfa diversidad (α-diversidad)</a>
  <ul class="collapse">
  <li><a href="#simpson-index" id="toc-simpson-index" class="nav-link" data-scroll-target="#simpson-index">Simpson Index</a></li>
  <li><a href="#fórmula-del-índice-de-simpson" id="toc-fórmula-del-índice-de-simpson" class="nav-link" data-scroll-target="#fórmula-del-índice-de-simpson">Fórmula del Índice de Simpson:</a></li>
  <li><a href="#interpretación" id="toc-interpretación" class="nav-link" data-scroll-target="#interpretación">Interpretación:</a></li>
  <li><a href="#shannon" id="toc-shannon" class="nav-link" data-scroll-target="#shannon">Shannon</a></li>
  <li><a href="#otras-métricas" id="toc-otras-métricas" class="nav-link" data-scroll-target="#otras-métricas">Otras métricas</a></li>
  <li><a href="#eveness-equitabilidad" id="toc-eveness-equitabilidad" class="nav-link" data-scroll-target="#eveness-equitabilidad">Eveness (equitabilidad)</a></li>
  <li><a href="#faiths-phylogenetic-diversity-pd" id="toc-faiths-phylogenetic-diversity-pd" class="nav-link" data-scroll-target="#faiths-phylogenetic-diversity-pd">Faith’s Phylogenetic Diversity (PD)</a></li>
  </ul></li>
  <li><a href="#curvas-de-rarefacción" id="toc-curvas-de-rarefacción" class="nav-link" data-scroll-target="#curvas-de-rarefacción">Curvas de rarefacción</a>
  <ul class="collapse">
  <li><a href="#observed-features" id="toc-observed-features" class="nav-link" data-scroll-target="#observed-features">Observed Features</a></li>
  <li><a href="#shannon-1" id="toc-shannon-1" class="nav-link" data-scroll-target="#shannon-1">Shannon</a></li>
  <li><a href="#faith-pd" id="toc-faith-pd" class="nav-link" data-scroll-target="#faith-pd">Faith PD</a></li>
  </ul></li>
  <li><a href="#beta-diversidad-β-diversidad" id="toc-beta-diversidad-β-diversidad" class="nav-link" data-scroll-target="#beta-diversidad-β-diversidad">Beta diversidad (β-diversidad)</a>
  <ul class="collapse">
  <li><a href="#distancia-bray-curtis" id="toc-distancia-bray-curtis" class="nav-link" data-scroll-target="#distancia-bray-curtis">Distancia Bray Curtis</a></li>
  <li><a href="#distancia-de-jaccard" id="toc-distancia-de-jaccard" class="nav-link" data-scroll-target="#distancia-de-jaccard">Distancia de Jaccard</a></li>
  <li><a href="#unifrac" id="toc-unifrac" class="nav-link" data-scroll-target="#unifrac">UniFrac</a></li>
  <li><a href="#unweighted-unifrac" id="toc-unweighted-unifrac" class="nav-link" data-scroll-target="#unweighted-unifrac">Unweighted UniFrac</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#taxonomía" id="toc-taxonomía" class="nav-link" data-scroll-target="#taxonomía">Taxonomía</a>
  <ul class="collapse">
  <li><a href="#diversidadabundancía" id="toc-diversidadabundancía" class="nav-link" data-scroll-target="#diversidadabundancía">Diversidad/abundancía</a></li>
  <li><a href="#pruebas-de-abundancia-diferencial-con-ancom-bc" id="toc-pruebas-de-abundancia-diferencial-con-ancom-bc" class="nav-link" data-scroll-target="#pruebas-de-abundancia-diferencial-con-ancom-bc">Pruebas de abundancia diferencial con ANCOM-BC</a>
  <ul class="collapse">
  <li><a href="#pasos-principales-de-ancom-bc" id="toc-pasos-principales-de-ancom-bc" class="nav-link" data-scroll-target="#pasos-principales-de-ancom-bc">Pasos Principales de ANCOM-BC:</a></li>
  <li><a href="#ecuación-general-de-log-fold-change-estimado" id="toc-ecuación-general-de-log-fold-change-estimado" class="nav-link" data-scroll-target="#ecuación-general-de-log-fold-change-estimado">Ecuación general de log-fold change estimado:</a></li>
  <li><a href="#interpretación-de-resultados" id="toc-interpretación-de-resultados" class="nav-link" data-scroll-target="#interpretación-de-resultados">Interpretación de resultados:</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#apéndice" id="toc-apéndice" class="nav-link" data-scroll-target="#apéndice">Apéndice</a>
  <ul class="collapse">
  <li><a href="#abrir-archivos-de-qiime2-en-r" id="toc-abrir-archivos-de-qiime2-en-r" class="nav-link" data-scroll-target="#abrir-archivos-de-qiime2-en-r">Abrir archivos de Qiime2 en R</a></li>
  <li><a href="#otras-métricas-de-alfa-diversidad" id="toc-otras-métricas-de-alfa-diversidad" class="nav-link" data-scroll-target="#otras-métricas-de-alfa-diversidad">Otras métricas de Alfa Diversidad</a>
  <ul class="collapse">
  <li><a href="#usando-r-para-gŕaficar" id="toc-usando-r-para-gŕaficar" class="nav-link" data-scroll-target="#usando-r-para-gŕaficar">Usando R para gŕaficar</a></li>
  <li><a href="#faith-pd-1" id="toc-faith-pd-1" class="nav-link" data-scroll-target="#faith-pd-1">Faith-PD</a></li>
  </ul></li>
  <li><a href="#fisher-alpha" id="toc-fisher-alpha" class="nav-link" data-scroll-target="#fisher-alpha">Fisher Alpha</a></li>
  <li><a href="#section" id="toc-section" class="nav-link" data-scroll-target="#section"></a></li>
  <li><a href="#otras-métricas-de-beta-diversidad" id="toc-otras-métricas-de-beta-diversidad" class="nav-link" data-scroll-target="#otras-métricas-de-beta-diversidad">Otras métricas de Beta Diversidad</a>
  <ul class="collapse">
  <li><a href="#usando-r-para-gráficar" id="toc-usando-r-para-gráficar" class="nav-link" data-scroll-target="#usando-r-para-gráficar">Usando R para gráficar</a></li>
  </ul></li>
  <li><a href="#entrenamiento-base-de-datos-silva" id="toc-entrenamiento-base-de-datos-silva" class="nav-link" data-scroll-target="#entrenamiento-base-de-datos-silva">Entrenamiento base de datos SILVA</a>
  <ul class="collapse">
  <li><a href="#descarga-de-la-base-de-datos-silva" id="toc-descarga-de-la-base-de-datos-silva" class="nav-link" data-scroll-target="#descarga-de-la-base-de-datos-silva">Descarga de la base de datos SILVA</a></li>
  <li><a href="#conversión-de-rna-a-dna" id="toc-conversión-de-rna-a-dna" class="nav-link" data-scroll-target="#conversión-de-rna-a-dna">Conversión de RNA a DNA</a></li>
  <li><a href="#eliminación-de-secuencias-de-baja-calidad" id="toc-eliminación-de-secuencias-de-baja-calidad" class="nav-link" data-scroll-target="#eliminación-de-secuencias-de-baja-calidad">Eliminación de secuencias de baja calidad</a></li>
  <li><a href="#filtrado-de-secuencias-por-longitud-y-taxón" id="toc-filtrado-de-secuencias-por-longitud-y-taxón" class="nav-link" data-scroll-target="#filtrado-de-secuencias-por-longitud-y-taxón">Filtrado de secuencias por longitud y taxón</a></li>
  <li><a href="#dereplicación-de-secuencias" id="toc-dereplicación-de-secuencias" class="nav-link" data-scroll-target="#dereplicación-de-secuencias">Dereplicación de secuencias</a></li>
  <li><a href="#extracción-de-la-región-v4" id="toc-extracción-de-la-región-v4" class="nav-link" data-scroll-target="#extracción-de-la-región-v4">Extracción de la región V4</a></li>
  <li><a href="#dereplicación-de-la-región-v4" id="toc-dereplicación-de-la-región-v4" class="nav-link" data-scroll-target="#dereplicación-de-la-región-v4">Dereplicación de la región V4</a></li>
  <li><a href="#entrenamiento-del-clasificador-naive-bayes" id="toc-entrenamiento-del-clasificador-naive-bayes" class="nav-link" data-scroll-target="#entrenamiento-del-clasificador-naive-bayes">Entrenamiento del clasificador Naive Bayes</a></li>
  </ul></li>
  <li><a href="#boxplots-para-algunos-clados-importantes" id="toc-boxplots-para-algunos-clados-importantes" class="nav-link" data-scroll-target="#boxplots-para-algunos-clados-importantes">Boxplots para algunos clados importantes</a>
  <ul class="collapse">
  <li><a href="#lysobacteraceae" id="toc-lysobacteraceae" class="nav-link" data-scroll-target="#lysobacteraceae"><strong><em>Lysobacteraceae</em></strong></a></li>
  <li><a href="#micropepsis" id="toc-micropepsis" class="nav-link" data-scroll-target="#micropepsis"><strong><em>Micropepsis</em></strong></a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Metabarcoding: Tomate 16S usando Qiime2</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Autor/a</div>
    <div class="quarto-title-meta-contents">
             <p>OMICs Analysis </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>

<nav id="TOC-body" role="doc-toc">
    <h2 id="toc-title">Contenido</h2>
   
  <ul>
  <li><a href="#introducción" id="toc-introducción">Introducción</a></li>
  <li><a href="#importar-datos-pareados-con-qiime2" id="toc-importar-datos-pareados-con-qiime2">Importar datos pareados con Qiime2</a>
  <ul>
  <li><a href="#cutadapt" id="toc-cutadapt">Cutadapt</a>
  <ul>
  <li><a href="#explicación-del-comando" id="toc-explicación-del-comando">Explicación del comando:</a></li>
  <li><a href="#explorar-los-archivos" id="toc-explorar-los-archivos">Explorar los archivos</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#filtrado-de-lecturas-con-dada2" id="toc-filtrado-de-lecturas-con-dada2">Filtrado de lecturas con DADA2</a></li>
  <li><a href="#medidas-de-diversidad" id="toc-medidas-de-diversidad">Medidas de diversidad</a>
  <ul>
  <li><a href="#generación-de-un-árbol-filogenético" id="toc-generación-de-un-árbol-filogenético">Generación de un árbol filogenético</a></li>
  <li><a href="#métricas-general-de-filogenética" id="toc-métricas-general-de-filogenética">Métricas general de filogenética</a></li>
  <li><a href="#alfa-diversidad-α-diversidad" id="toc-alfa-diversidad-α-diversidad">Alfa diversidad (α-diversidad)</a>
  <ul>
  <li><a href="#simpson-index" id="toc-simpson-index">Simpson Index</a></li>
  <li><a href="#fórmula-del-índice-de-simpson" id="toc-fórmula-del-índice-de-simpson">Fórmula del Índice de Simpson:</a></li>
  <li><a href="#interpretación" id="toc-interpretación">Interpretación:</a></li>
  <li><a href="#shannon" id="toc-shannon">Shannon</a></li>
  <li><a href="#otras-métricas" id="toc-otras-métricas">Otras métricas</a></li>
  <li><a href="#eveness-equitabilidad" id="toc-eveness-equitabilidad">Eveness (equitabilidad)</a></li>
  <li><a href="#faiths-phylogenetic-diversity-pd" id="toc-faiths-phylogenetic-diversity-pd">Faith’s Phylogenetic Diversity (PD)</a></li>
  </ul></li>
  <li><a href="#curvas-de-rarefacción" id="toc-curvas-de-rarefacción">Curvas de rarefacción</a>
  <ul>
  <li><a href="#observed-features" id="toc-observed-features">Observed Features</a></li>
  <li><a href="#shannon-1" id="toc-shannon-1">Shannon</a></li>
  <li><a href="#faith-pd" id="toc-faith-pd">Faith PD</a></li>
  </ul></li>
  <li><a href="#beta-diversidad-β-diversidad" id="toc-beta-diversidad-β-diversidad">Beta diversidad (β-diversidad)</a>
  <ul>
  <li><a href="#distancia-bray-curtis" id="toc-distancia-bray-curtis">Distancia Bray Curtis</a></li>
  <li><a href="#distancia-de-jaccard" id="toc-distancia-de-jaccard">Distancia de Jaccard</a></li>
  <li><a href="#unifrac" id="toc-unifrac">UniFrac</a></li>
  <li><a href="#unweighted-unifrac" id="toc-unweighted-unifrac">Unweighted UniFrac</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#taxonomía" id="toc-taxonomía">Taxonomía</a>
  <ul>
  <li><a href="#diversidadabundancía" id="toc-diversidadabundancía">Diversidad/abundancía</a></li>
  <li><a href="#pruebas-de-abundancia-diferencial-con-ancom-bc" id="toc-pruebas-de-abundancia-diferencial-con-ancom-bc">Pruebas de abundancia diferencial con ANCOM-BC</a>
  <ul>
  <li><a href="#pasos-principales-de-ancom-bc" id="toc-pasos-principales-de-ancom-bc">Pasos Principales de ANCOM-BC:</a></li>
  <li><a href="#ecuación-general-de-log-fold-change-estimado" id="toc-ecuación-general-de-log-fold-change-estimado">Ecuación general de log-fold change estimado:</a></li>
  <li><a href="#interpretación-de-resultados" id="toc-interpretación-de-resultados">Interpretación de resultados:</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#apéndice" id="toc-apéndice">Apéndice</a>
  <ul>
  <li><a href="#abrir-archivos-de-qiime2-en-r" id="toc-abrir-archivos-de-qiime2-en-r">Abrir archivos de Qiime2 en R</a></li>
  <li><a href="#otras-métricas-de-alfa-diversidad" id="toc-otras-métricas-de-alfa-diversidad">Otras métricas de Alfa Diversidad</a>
  <ul>
  <li><a href="#usando-r-para-gŕaficar" id="toc-usando-r-para-gŕaficar">Usando R para gŕaficar</a></li>
  <li><a href="#faith-pd-1" id="toc-faith-pd-1">Faith-PD</a></li>
  </ul></li>
  <li><a href="#fisher-alpha" id="toc-fisher-alpha">Fisher Alpha</a></li>
  <li><a href="#section" id="toc-section"></a></li>
  <li><a href="#otras-métricas-de-beta-diversidad" id="toc-otras-métricas-de-beta-diversidad">Otras métricas de Beta Diversidad</a>
  <ul>
  <li><a href="#usando-r-para-gráficar" id="toc-usando-r-para-gráficar">Usando R para gráficar</a></li>
  </ul></li>
  <li><a href="#entrenamiento-base-de-datos-silva" id="toc-entrenamiento-base-de-datos-silva">Entrenamiento base de datos SILVA</a>
  <ul>
  <li><a href="#descarga-de-la-base-de-datos-silva" id="toc-descarga-de-la-base-de-datos-silva">Descarga de la base de datos SILVA</a></li>
  <li><a href="#conversión-de-rna-a-dna" id="toc-conversión-de-rna-a-dna">Conversión de RNA a DNA</a></li>
  <li><a href="#eliminación-de-secuencias-de-baja-calidad" id="toc-eliminación-de-secuencias-de-baja-calidad">Eliminación de secuencias de baja calidad</a></li>
  <li><a href="#filtrado-de-secuencias-por-longitud-y-taxón" id="toc-filtrado-de-secuencias-por-longitud-y-taxón">Filtrado de secuencias por longitud y taxón</a></li>
  <li><a href="#dereplicación-de-secuencias" id="toc-dereplicación-de-secuencias">Dereplicación de secuencias</a></li>
  <li><a href="#extracción-de-la-región-v4" id="toc-extracción-de-la-región-v4">Extracción de la región V4</a></li>
  <li><a href="#dereplicación-de-la-región-v4" id="toc-dereplicación-de-la-región-v4">Dereplicación de la región V4</a></li>
  <li><a href="#entrenamiento-del-clasificador-naive-bayes" id="toc-entrenamiento-del-clasificador-naive-bayes">Entrenamiento del clasificador Naive Bayes</a></li>
  </ul></li>
  <li><a href="#boxplots-para-algunos-clados-importantes" id="toc-boxplots-para-algunos-clados-importantes">Boxplots para algunos clados importantes</a>
  <ul>
  <li><a href="#lysobacteraceae" id="toc-lysobacteraceae"><strong><em>Lysobacteraceae</em></strong></a></li>
  <li><a href="#micropepsis" id="toc-micropepsis"><strong><em>Micropepsis</em></strong></a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/logo_OAN_for_white.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="300"></p>
</figure>
</div>
<section id="introducción" class="level1">
<h1>Introducción</h1>
<p>Datos de partida “<a href="https://www.ncbi.nlm.nih.gov/sra/?term=SRX13358039">Disentangling the genetic basis of rhizosphere microbiome assembly in tomato</a>”</p>
<p>Para este tutorial vamos a utilizar <a href="https://qiime2.org/">Qiime2</a><strong>,</strong> <a href="https://library.qiime2.org/quickstart">aquí</a> lo puedes instalar según sea tu sistema operativo. Para los siguientes pasos vamos a autilizar la distribución “<a href="https://amplicon-docs.qiime2.org/">amplicon</a>”. Este tutorial se hizo utilizando el sistema operativo <code>Ubuntu 22.04.5 LTS</code>, en un equipo con <code>24 Gb</code> de <code>RAM</code> <code>Intel® Core™ i7-3610QM CPU @ 2.30GHz × 8</code>. Todo se hizo usando la <code>terminal</code>.</p>
<p>Una vez instalado <code>qiime</code> activa el ambiente conda.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate qiime2-amplicon-2024.10</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Primer paso:</strong> Genera una carpeta (directorio) exclusiva para llevar a cabo el proyecto, en este caso el nombre de la carpeta es <code>2025_Demo_tomato_16S</code>, en ella se va generar una carpeta para guardar los genomas 16S que tendrá por nombre <code>data</code>.</p>
<p>Lo genomas 16S los podrás encontrar <a href="https://drive.google.com/drive/folders/1mQRMfCqHzCaF9_sc_az1Cj3l6YCAAB5V?usp=drive_link"><strong>aquí</strong></a> para descargar.</p>
<p>Escribe los siguiente comando en la terminal para crear el directorio.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> 2025_Demo_tomato_16S</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> 2025_Demo_tomato_16S</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> data</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="importar-datos-pareados-con-qiime2" class="level1">
<h1>Importar datos pareados con Qiime2</h1>
<p>Vamos a generar un archivo <code>manifiesto</code> en el vamos a indicar la localización de las lecturas pareadas. El archivo se ve de la siguiente manera:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>   sample.id      forward.absolute.filepath      reverse.absolute.filepath
1   Tomato_1 $PWD/data/Tomato_1_R1.fastq.gz $PWD/data/Tomato_1_R2.fastq.gz
2   Tomato_2 $PWD/data/Tomato_2_R1.fastq.gz $PWD/data/Tomato_2_R2.fastq.gz
3   Tomato_3 $PWD/data/Tomato_3_R1.fastq.gz $PWD/data/Tomato_3_R2.fastq.gz
4   Tomato_4 $PWD/data/Tomato_4_R1.fastq.gz $PWD/data/Tomato_4_R2.fastq.gz
5   Tomato_5 $PWD/data/Tomato_5_R1.fastq.gz $PWD/data/Tomato_5_R2.fastq.gz
6   Tomato_6 $PWD/data/Tomato_6_R1.fastq.gz $PWD/data/Tomato_6_R2.fastq.gz
7       NC_1     $PWD/data/NC_1_R1.fastq.gz     $PWD/data/NC_1_R2.fastq.gz
8       NC_2     $PWD/data/NC_2_R1.fastq.gz     $PWD/data/NC_2_R2.fastq.gz
9       NC_3     $PWD/data/NC_3_R1.fastq.gz     $PWD/data/NC_3_R2.fastq.gz
10      NC_4     $PWD/data/NC_4_R1.fastq.gz     $PWD/data/NC_4_R2.fastq.gz
11      NC_5     $PWD/data/NC_5_R1.fastq.gz     $PWD/data/NC_5_R2.fastq.gz
12      NC_6     $PWD/data/NC_6_R1.fastq.gz     $PWD/data/NC_6_R2.fastq.gz</code></pre>
</div>
</div>
<p>Este archivo lo vamos a nombrar como <code>tomato_manifest</code>, este archivo es de tipo <code>tab-separate-value</code>. Puedes descargar el archivo <code>manifiesto</code> <a href="https://drive.google.com/file/d/1hIzzApEfZfKqOlxhjVDsSvKVmqUbEA1d/view?usp=drive_link">aquí</a>.</p>
<p><strong>NOTA</strong> ⚠️<strong>: Durante todo este tutorial vamos a ir generando dos clases de archivos</strong></p>
<ul>
<li><p><code>.qza</code> <strong>QIIME Zipped Artifact</strong></p></li>
<li><p><code>.qzv</code> <strong>QIIME Zipped Visualization</strong></p></li>
</ul>
<p>Para ver todos los archivos <code>.qzv</code> basta con utilizar el comando <code>qiime tools view</code> seguido del nombre de tu archivo. Más adelante lo verás más claro.</p>
<p>Para importar los secuencias 16S pareadas vamos a utilizar el siguiente código:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools import <span class="dt">\</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>--type <span class="st">'SampleData[PairedEndSequencesWithQuality]'</span> <span class="dt">\</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>--input-path tomato_manifest <span class="dt">\</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>--output-path paired-end-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>--input-format PairedEndFastqManifestPhred33V2</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Para visualizar la calidad de los archivos importado podemos usar el siguiente comando:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> demux summarize <span class="dt">\</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>--i-data paired-end-tomato-16s.qza <span class="dt">\ </span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="ex">--o-visualization</span> summary_tmp_tomato_16s.qzv</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view summary_tmp_tomato_16s.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/forwdard_q1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/reverse_q1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p><strong>Número de lecturas por muestra</strong></p>
<div class="cell">
<div class="cell-output-display">
<div id="aruvcronbn" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#aruvcronbn table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#aruvcronbn thead, #aruvcronbn tbody, #aruvcronbn tfoot, #aruvcronbn tr, #aruvcronbn td, #aruvcronbn th {
  border-style: none;
}

#aruvcronbn p {
  margin: 0;
  padding: 0;
}

#aruvcronbn .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#aruvcronbn .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#aruvcronbn .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#aruvcronbn .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#aruvcronbn .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#aruvcronbn .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aruvcronbn .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#aruvcronbn .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#aruvcronbn .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#aruvcronbn .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#aruvcronbn .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#aruvcronbn .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#aruvcronbn .gt_spanner_row {
  border-bottom-style: hidden;
}

#aruvcronbn .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#aruvcronbn .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#aruvcronbn .gt_from_md > :first-child {
  margin-top: 0;
}

#aruvcronbn .gt_from_md > :last-child {
  margin-bottom: 0;
}

#aruvcronbn .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#aruvcronbn .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#aruvcronbn .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#aruvcronbn .gt_row_group_first td {
  border-top-width: 2px;
}

#aruvcronbn .gt_row_group_first th {
  border-top-width: 2px;
}

#aruvcronbn .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#aruvcronbn .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#aruvcronbn .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#aruvcronbn .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aruvcronbn .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#aruvcronbn .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#aruvcronbn .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#aruvcronbn .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#aruvcronbn .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aruvcronbn .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#aruvcronbn .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#aruvcronbn .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#aruvcronbn .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#aruvcronbn .gt_left {
  text-align: left;
}

#aruvcronbn .gt_center {
  text-align: center;
}

#aruvcronbn .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#aruvcronbn .gt_font_normal {
  font-weight: normal;
}

#aruvcronbn .gt_font_bold {
  font-weight: bold;
}

#aruvcronbn .gt_font_italic {
  font-style: italic;
}

#aruvcronbn .gt_super {
  font-size: 65%;
}

#aruvcronbn .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#aruvcronbn .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#aruvcronbn .gt_indent_1 {
  text-indent: 5px;
}

#aruvcronbn .gt_indent_2 {
  text-indent: 10px;
}

#aruvcronbn .gt_indent_3 {
  text-indent: 15px;
}

#aruvcronbn .gt_indent_4 {
  text-indent: 20px;
}

#aruvcronbn .gt_indent_5 {
  text-indent: 25px;
}

#aruvcronbn .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#aruvcronbn div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="sample-ID" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">sample ID</th>
<th id="forward-sequence-count" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">forward sequence count</th>
<th id="reverse-sequence-count" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">reverse sequence count</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="sample ID">NC_1</td>
<td class="gt_row gt_right" headers="forward sequence count">45207</td>
<td class="gt_row gt_right" headers="reverse sequence count">45207</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample ID">NC_2</td>
<td class="gt_row gt_right" headers="forward sequence count">42412</td>
<td class="gt_row gt_right" headers="reverse sequence count">42412</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample ID">NC_3</td>
<td class="gt_row gt_right" headers="forward sequence count">39468</td>
<td class="gt_row gt_right" headers="reverse sequence count">39468</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample ID">NC_4</td>
<td class="gt_row gt_right" headers="forward sequence count">40529</td>
<td class="gt_row gt_right" headers="reverse sequence count">40529</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample ID">NC_5</td>
<td class="gt_row gt_right" headers="forward sequence count">41755</td>
<td class="gt_row gt_right" headers="reverse sequence count">41755</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample ID">NC_6</td>
<td class="gt_row gt_right" headers="forward sequence count">40153</td>
<td class="gt_row gt_right" headers="reverse sequence count">40153</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample ID">Tomato_1</td>
<td class="gt_row gt_right" headers="forward sequence count">34546</td>
<td class="gt_row gt_right" headers="reverse sequence count">34546</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample ID">Tomato_2</td>
<td class="gt_row gt_right" headers="forward sequence count">41028</td>
<td class="gt_row gt_right" headers="reverse sequence count">41028</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample ID">Tomato_3</td>
<td class="gt_row gt_right" headers="forward sequence count">40793</td>
<td class="gt_row gt_right" headers="reverse sequence count">40793</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample ID">Tomato_4</td>
<td class="gt_row gt_right" headers="forward sequence count">39477</td>
<td class="gt_row gt_right" headers="reverse sequence count">39477</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample ID">Tomato_5</td>
<td class="gt_row gt_right" headers="forward sequence count">33744</td>
<td class="gt_row gt_right" headers="reverse sequence count">33744</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample ID">Tomato_6</td>
<td class="gt_row gt_right" headers="forward sequence count">37507</td>
<td class="gt_row gt_right" headers="reverse sequence count">37507</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<section id="cutadapt" class="level2">
<h2 class="anchored" data-anchor-id="cutadapt">Cutadapt</h2>
<p>Una vez importadas las secuencias, vamos a realizar un preprocesmaiento con <a href="https://cutadapt.readthedocs.io/en/stable/"><code>cutadapt</code></a> para eliminar adapdores, <code>qiime2</code> nos permite hacer esto utilizando la siguiente instrucción:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> cutadapt trim-paired <span class="dt">\</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>--i-demultiplexed-sequences paired-end-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>--p-front-f CCTACGGGNGGCWGCAG <span class="dt">\</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>--p-adapter-f GGATTAGATACCCBDGTAGTC <span class="dt">\</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>--p-front-r GACTACHVGGGTATCTAATCC <span class="dt">\</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>--p-adapter-r CTGCWGCCNCCCGTAGG <span class="dt">\</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>--p-match-read-wildcards <span class="dt">\</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>--p-discard-untrimmed <span class="dt">\</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>--o-trimmed-sequences trimmed_paired-end-tomato-16s.qza <span class="at">--p-cores</span> 5</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="explicación-del-comando" class="level3">
<h3 class="anchored" data-anchor-id="explicación-del-comando">Explicación del comando:</h3>
<ul>
<li><p><strong><code>--i-demultiplexed-sequences paired-end-tomato-16s.qza</code></strong><br>
</p>
<p>Este parámetro indica el archivo de entrada que contiene tus secuencias pareadas ya demultiplexadas. El archivo de entrada es <code>paired-end-tomato-16s.qza</code>.</p></li>
<li><p><strong>Adaptadores para las lecturas forward y reverse</strong></p>
<ul>
<li><p><strong><code>--p-front-f CCTACGGGNGGCWGCAG</code></strong>: Adaptador 5’ de las lecturas forward.</p></li>
<li><p><strong><code>--p-adapter-f GGATTAGATACCCBDGTAGTC</code></strong>: Adaptador 3’ de las lecturas forward.</p></li>
<li><p><strong><code>--p-front-r GACTACHVGGGTATCTAATCC</code></strong>: Adaptador 5’ de las lecturas reverse.</p></li>
<li><p><strong><code>--p-adapter-r CTGCWGCCNCCCGTAGG</code></strong>: Adaptador 3’ de las lecturas reverse.</p></li>
</ul></li>
<li><p><strong><code>--p-match-read-wildcards</code></strong><br>
</p>
<p>Este flag permite que los comodines (como N, W, B, D, H, V) en los adaptadores coincidan con las bases en las secuencias de lectura.</p></li>
<li><p><strong><code>--p-discard-untrimmed</code></strong><br>
</p>
<p>Esta opción descarta las secuencias que no contienen los adaptadores especificados, asegurando que solo se conserven las lecturas correctamente recortadas.</p></li>
<li><p><strong><code>--o-trimmed-sequences trimmed.qza</code></strong><br>
</p>
<p>Define el archivo de salida en formato <code>QZA</code> (<code>trimmed_paired-end-tomato-16s.qza</code>) que contendrá las secuencias recortadas. <code>Qiime</code> genera un solo archivo <code>QZA</code> que puedes usar en pasos posteriores del análisis.</p></li>
</ul>
</section>
<section id="explorar-los-archivos" class="level3">
<h3 class="anchored" data-anchor-id="explorar-los-archivos">Explorar los archivos</h3>
<p>Qiime permite generar un reporte HTML para ver la cantidad de lecturas por muestras y métricas generales de secuencia. Para ello podemos generar el siguiente archivo:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> demux summarize <span class="dt">\</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>--i-data trimmed_paired-end-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>--o-visualization summary_tomato_16s.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Para visualizar este archivo utiliza el siguiente código:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view summary_tomato_16s.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Número de cuentas por muestras después de cuatadapt</strong></p>
<div class="cell">
<div class="cell-output-display">
<div id="fwvmlsqhwf" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#fwvmlsqhwf table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#fwvmlsqhwf thead, #fwvmlsqhwf tbody, #fwvmlsqhwf tfoot, #fwvmlsqhwf tr, #fwvmlsqhwf td, #fwvmlsqhwf th {
  border-style: none;
}

#fwvmlsqhwf p {
  margin: 0;
  padding: 0;
}

#fwvmlsqhwf .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#fwvmlsqhwf .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#fwvmlsqhwf .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#fwvmlsqhwf .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#fwvmlsqhwf .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#fwvmlsqhwf .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fwvmlsqhwf .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#fwvmlsqhwf .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#fwvmlsqhwf .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#fwvmlsqhwf .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#fwvmlsqhwf .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#fwvmlsqhwf .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#fwvmlsqhwf .gt_spanner_row {
  border-bottom-style: hidden;
}

#fwvmlsqhwf .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#fwvmlsqhwf .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#fwvmlsqhwf .gt_from_md > :first-child {
  margin-top: 0;
}

#fwvmlsqhwf .gt_from_md > :last-child {
  margin-bottom: 0;
}

#fwvmlsqhwf .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#fwvmlsqhwf .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#fwvmlsqhwf .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#fwvmlsqhwf .gt_row_group_first td {
  border-top-width: 2px;
}

#fwvmlsqhwf .gt_row_group_first th {
  border-top-width: 2px;
}

#fwvmlsqhwf .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#fwvmlsqhwf .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#fwvmlsqhwf .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#fwvmlsqhwf .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fwvmlsqhwf .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#fwvmlsqhwf .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#fwvmlsqhwf .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#fwvmlsqhwf .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#fwvmlsqhwf .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fwvmlsqhwf .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#fwvmlsqhwf .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#fwvmlsqhwf .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#fwvmlsqhwf .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#fwvmlsqhwf .gt_left {
  text-align: left;
}

#fwvmlsqhwf .gt_center {
  text-align: center;
}

#fwvmlsqhwf .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#fwvmlsqhwf .gt_font_normal {
  font-weight: normal;
}

#fwvmlsqhwf .gt_font_bold {
  font-weight: bold;
}

#fwvmlsqhwf .gt_font_italic {
  font-style: italic;
}

#fwvmlsqhwf .gt_super {
  font-size: 65%;
}

#fwvmlsqhwf .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#fwvmlsqhwf .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#fwvmlsqhwf .gt_indent_1 {
  text-indent: 5px;
}

#fwvmlsqhwf .gt_indent_2 {
  text-indent: 10px;
}

#fwvmlsqhwf .gt_indent_3 {
  text-indent: 15px;
}

#fwvmlsqhwf .gt_indent_4 {
  text-indent: 20px;
}

#fwvmlsqhwf .gt_indent_5 {
  text-indent: 25px;
}

#fwvmlsqhwf .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#fwvmlsqhwf div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="sample-ID" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">sample ID</th>
<th id="forward-sequence-count" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">forward sequence count</th>
<th id="reverse-sequence-count" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">reverse sequence count</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="sample ID">NC_1</td>
<td class="gt_row gt_right" headers="forward sequence count">44773</td>
<td class="gt_row gt_right" headers="reverse sequence count">44773</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample ID">NC_2</td>
<td class="gt_row gt_right" headers="forward sequence count">41735</td>
<td class="gt_row gt_right" headers="reverse sequence count">41735</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample ID">NC_3</td>
<td class="gt_row gt_right" headers="forward sequence count">39165</td>
<td class="gt_row gt_right" headers="reverse sequence count">39165</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample ID">NC_4</td>
<td class="gt_row gt_right" headers="forward sequence count">40180</td>
<td class="gt_row gt_right" headers="reverse sequence count">40180</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample ID">NC_5</td>
<td class="gt_row gt_right" headers="forward sequence count">41342</td>
<td class="gt_row gt_right" headers="reverse sequence count">41342</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample ID">NC_6</td>
<td class="gt_row gt_right" headers="forward sequence count">39829</td>
<td class="gt_row gt_right" headers="reverse sequence count">39829</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample ID">Tomato_1</td>
<td class="gt_row gt_right" headers="forward sequence count">34194</td>
<td class="gt_row gt_right" headers="reverse sequence count">34194</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample ID">Tomato_2</td>
<td class="gt_row gt_right" headers="forward sequence count">40436</td>
<td class="gt_row gt_right" headers="reverse sequence count">40436</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample ID">Tomato_3</td>
<td class="gt_row gt_right" headers="forward sequence count">40454</td>
<td class="gt_row gt_right" headers="reverse sequence count">40454</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample ID">Tomato_4</td>
<td class="gt_row gt_right" headers="forward sequence count">39106</td>
<td class="gt_row gt_right" headers="reverse sequence count">39106</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample ID">Tomato_5</td>
<td class="gt_row gt_right" headers="forward sequence count">33238</td>
<td class="gt_row gt_right" headers="reverse sequence count">33238</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample ID">Tomato_6</td>
<td class="gt_row gt_right" headers="forward sequence count">37085</td>
<td class="gt_row gt_right" headers="reverse sequence count">37085</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
</section>
</section>
<section id="filtrado-de-lecturas-con-dada2" class="level1">
<h1>Filtrado de lecturas con <a href="https://benjjneb.github.io/dada2/">DADA2</a></h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> dada2 denoise-paired <span class="dt">\</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>--i-demultiplexed-seqs trimmed_paired-end-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>--p-trunc-len-f 280 <span class="dt">\</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>--p-trunc-len-r 260 <span class="dt">\</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>--o-representative-sequences rep-seqs-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>--o-table table-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>--o-denoising-stats stats-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>--p-n-threads 5</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Con esto podemos ver un reporte de la cantidad de lecturas filtradas:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> metadata tabulate <span class="dt">\</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--m-input-file</span> stats-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> stats-dada2-tomato-16s.qzv</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view stats-dada2-tomato-16s.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Lecturas filtradas con DADA2</strong></p>
<div class="cell">
<div class="cell-output-display">
<div id="ebcdscvydf" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#ebcdscvydf table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#ebcdscvydf thead, #ebcdscvydf tbody, #ebcdscvydf tfoot, #ebcdscvydf tr, #ebcdscvydf td, #ebcdscvydf th {
  border-style: none;
}

#ebcdscvydf p {
  margin: 0;
  padding: 0;
}

#ebcdscvydf .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#ebcdscvydf .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#ebcdscvydf .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#ebcdscvydf .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#ebcdscvydf .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ebcdscvydf .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ebcdscvydf .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ebcdscvydf .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#ebcdscvydf .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#ebcdscvydf .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#ebcdscvydf .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#ebcdscvydf .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#ebcdscvydf .gt_spanner_row {
  border-bottom-style: hidden;
}

#ebcdscvydf .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#ebcdscvydf .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#ebcdscvydf .gt_from_md > :first-child {
  margin-top: 0;
}

#ebcdscvydf .gt_from_md > :last-child {
  margin-bottom: 0;
}

#ebcdscvydf .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#ebcdscvydf .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#ebcdscvydf .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#ebcdscvydf .gt_row_group_first td {
  border-top-width: 2px;
}

#ebcdscvydf .gt_row_group_first th {
  border-top-width: 2px;
}

#ebcdscvydf .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ebcdscvydf .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#ebcdscvydf .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#ebcdscvydf .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ebcdscvydf .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ebcdscvydf .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#ebcdscvydf .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#ebcdscvydf .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#ebcdscvydf .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ebcdscvydf .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ebcdscvydf .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ebcdscvydf .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ebcdscvydf .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ebcdscvydf .gt_left {
  text-align: left;
}

#ebcdscvydf .gt_center {
  text-align: center;
}

#ebcdscvydf .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#ebcdscvydf .gt_font_normal {
  font-weight: normal;
}

#ebcdscvydf .gt_font_bold {
  font-weight: bold;
}

#ebcdscvydf .gt_font_italic {
  font-style: italic;
}

#ebcdscvydf .gt_super {
  font-size: 65%;
}

#ebcdscvydf .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#ebcdscvydf .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#ebcdscvydf .gt_indent_1 {
  text-indent: 5px;
}

#ebcdscvydf .gt_indent_2 {
  text-indent: 10px;
}

#ebcdscvydf .gt_indent_3 {
  text-indent: 15px;
}

#ebcdscvydf .gt_indent_4 {
  text-indent: 20px;
}

#ebcdscvydf .gt_indent_5 {
  text-indent: 25px;
}

#ebcdscvydf .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#ebcdscvydf div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="sample-id" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">sample-id</th>
<th id="input" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">input</th>
<th id="filtered" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">filtered</th>
<th id="percentage-of-input-passed-filter" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">percentage of input passed filter</th>
<th id="denoised" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">denoised</th>
<th id="merged" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">merged</th>
<th id="percentage-of-input-merged" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">percentage of input merged</th>
<th id="non-chimeric" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">non-chimeric</th>
<th id="percentage-of-input-non-chimeric" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">percentage of input non-chimeric</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="sample-id">#q2:types</td>
<td class="gt_row gt_left" headers="input">numeric</td>
<td class="gt_row gt_left" headers="filtered">numeric</td>
<td class="gt_row gt_left" headers="percentage of input passed filter">numeric</td>
<td class="gt_row gt_left" headers="denoised">numeric</td>
<td class="gt_row gt_left" headers="merged">numeric</td>
<td class="gt_row gt_left" headers="percentage of input merged">numeric</td>
<td class="gt_row gt_left" headers="non-chimeric">numeric</td>
<td class="gt_row gt_left" headers="percentage of input non-chimeric">numeric</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample-id">NC_1</td>
<td class="gt_row gt_left" headers="input">44773</td>
<td class="gt_row gt_left" headers="filtered">33553</td>
<td class="gt_row gt_left" headers="percentage of input passed filter">74.94</td>
<td class="gt_row gt_left" headers="denoised">27791</td>
<td class="gt_row gt_left" headers="merged">21557</td>
<td class="gt_row gt_left" headers="percentage of input merged">48.15</td>
<td class="gt_row gt_left" headers="non-chimeric">21086</td>
<td class="gt_row gt_left" headers="percentage of input non-chimeric">47.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample-id">NC_2</td>
<td class="gt_row gt_left" headers="input">41735</td>
<td class="gt_row gt_left" headers="filtered">30374</td>
<td class="gt_row gt_left" headers="percentage of input passed filter">72.78</td>
<td class="gt_row gt_left" headers="denoised">25195</td>
<td class="gt_row gt_left" headers="merged">19560</td>
<td class="gt_row gt_left" headers="percentage of input merged">46.87</td>
<td class="gt_row gt_left" headers="non-chimeric">19109</td>
<td class="gt_row gt_left" headers="percentage of input non-chimeric">45.79</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample-id">NC_3</td>
<td class="gt_row gt_left" headers="input">39165</td>
<td class="gt_row gt_left" headers="filtered">30886</td>
<td class="gt_row gt_left" headers="percentage of input passed filter">78.86</td>
<td class="gt_row gt_left" headers="denoised">25397</td>
<td class="gt_row gt_left" headers="merged">19667</td>
<td class="gt_row gt_left" headers="percentage of input merged">50.22</td>
<td class="gt_row gt_left" headers="non-chimeric">19200</td>
<td class="gt_row gt_left" headers="percentage of input non-chimeric">49.02</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample-id">NC_4</td>
<td class="gt_row gt_left" headers="input">40180</td>
<td class="gt_row gt_left" headers="filtered">31675</td>
<td class="gt_row gt_left" headers="percentage of input passed filter">78.83</td>
<td class="gt_row gt_left" headers="denoised">26583</td>
<td class="gt_row gt_left" headers="merged">20699</td>
<td class="gt_row gt_left" headers="percentage of input merged">51.52</td>
<td class="gt_row gt_left" headers="non-chimeric">20335</td>
<td class="gt_row gt_left" headers="percentage of input non-chimeric">50.61</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample-id">NC_5</td>
<td class="gt_row gt_left" headers="input">41342</td>
<td class="gt_row gt_left" headers="filtered">32540</td>
<td class="gt_row gt_left" headers="percentage of input passed filter">78.71</td>
<td class="gt_row gt_left" headers="denoised">27628</td>
<td class="gt_row gt_left" headers="merged">21835</td>
<td class="gt_row gt_left" headers="percentage of input merged">52.82</td>
<td class="gt_row gt_left" headers="non-chimeric">21225</td>
<td class="gt_row gt_left" headers="percentage of input non-chimeric">51.34</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample-id">NC_6</td>
<td class="gt_row gt_left" headers="input">39829</td>
<td class="gt_row gt_left" headers="filtered">31863</td>
<td class="gt_row gt_left" headers="percentage of input passed filter">80</td>
<td class="gt_row gt_left" headers="denoised">27612</td>
<td class="gt_row gt_left" headers="merged">22583</td>
<td class="gt_row gt_left" headers="percentage of input merged">56.7</td>
<td class="gt_row gt_left" headers="non-chimeric">20297</td>
<td class="gt_row gt_left" headers="percentage of input non-chimeric">50.96</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample-id">Tomato_1</td>
<td class="gt_row gt_left" headers="input">34194</td>
<td class="gt_row gt_left" headers="filtered">27222</td>
<td class="gt_row gt_left" headers="percentage of input passed filter">79.61</td>
<td class="gt_row gt_left" headers="denoised">23865</td>
<td class="gt_row gt_left" headers="merged">19237</td>
<td class="gt_row gt_left" headers="percentage of input merged">56.26</td>
<td class="gt_row gt_left" headers="non-chimeric">17999</td>
<td class="gt_row gt_left" headers="percentage of input non-chimeric">52.64</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample-id">Tomato_2</td>
<td class="gt_row gt_left" headers="input">40436</td>
<td class="gt_row gt_left" headers="filtered">30780</td>
<td class="gt_row gt_left" headers="percentage of input passed filter">76.12</td>
<td class="gt_row gt_left" headers="denoised">26517</td>
<td class="gt_row gt_left" headers="merged">20876</td>
<td class="gt_row gt_left" headers="percentage of input merged">51.63</td>
<td class="gt_row gt_left" headers="non-chimeric">19544</td>
<td class="gt_row gt_left" headers="percentage of input non-chimeric">48.33</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample-id">Tomato_3</td>
<td class="gt_row gt_left" headers="input">40454</td>
<td class="gt_row gt_left" headers="filtered">32515</td>
<td class="gt_row gt_left" headers="percentage of input passed filter">80.38</td>
<td class="gt_row gt_left" headers="denoised">28693</td>
<td class="gt_row gt_left" headers="merged">23733</td>
<td class="gt_row gt_left" headers="percentage of input merged">58.67</td>
<td class="gt_row gt_left" headers="non-chimeric">20976</td>
<td class="gt_row gt_left" headers="percentage of input non-chimeric">51.85</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample-id">Tomato_4</td>
<td class="gt_row gt_left" headers="input">39106</td>
<td class="gt_row gt_left" headers="filtered">30974</td>
<td class="gt_row gt_left" headers="percentage of input passed filter">79.21</td>
<td class="gt_row gt_left" headers="denoised">26607</td>
<td class="gt_row gt_left" headers="merged">20830</td>
<td class="gt_row gt_left" headers="percentage of input merged">53.27</td>
<td class="gt_row gt_left" headers="non-chimeric">19951</td>
<td class="gt_row gt_left" headers="percentage of input non-chimeric">51.02</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="sample-id">Tomato_5</td>
<td class="gt_row gt_left" headers="input">33238</td>
<td class="gt_row gt_left" headers="filtered">26701</td>
<td class="gt_row gt_left" headers="percentage of input passed filter">80.33</td>
<td class="gt_row gt_left" headers="denoised">22755</td>
<td class="gt_row gt_left" headers="merged">17761</td>
<td class="gt_row gt_left" headers="percentage of input merged">53.44</td>
<td class="gt_row gt_left" headers="non-chimeric">16893</td>
<td class="gt_row gt_left" headers="percentage of input non-chimeric">50.82</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="sample-id">Tomato_6</td>
<td class="gt_row gt_left" headers="input">37085</td>
<td class="gt_row gt_left" headers="filtered">30045</td>
<td class="gt_row gt_left" headers="percentage of input passed filter">81.02</td>
<td class="gt_row gt_left" headers="denoised">26154</td>
<td class="gt_row gt_left" headers="merged">20467</td>
<td class="gt_row gt_left" headers="percentage of input merged">55.19</td>
<td class="gt_row gt_left" headers="non-chimeric">19112</td>
<td class="gt_row gt_left" headers="percentage of input non-chimeric">51.54</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<p>Después de terminar el paso de filtrado de calidad, querrás explorar los datos que obtuviste. Este filtrado es importante porque elimina las secuencias de baja calidad o con errores, dejándote datos más confiables para trabajar. Para revisar esos datos, puedes usar dos comandos que te muestran resúmenes visuales fáciles de entender.</p>
<p>Los datos asociados los vamos a depositar en un archivo nombrado <code>Experiment_Design.tsv</code>:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 13 × 4
   `Sample-id` accession   sra_run     Type       
   &lt;chr&gt;       &lt;chr&gt;       &lt;chr&gt;       &lt;chr&gt;      
 1 #q2:types   categorical categorical categorical
 2 Tomato_1    SRX13358040 SRR17174837 Tomato     
 3 Tomato_2    SRX13358039 SRR17174838 Tomato     
 4 Tomato_3    SRX13358038 SRR17174839 Tomato     
 5 Tomato_4    SRX13358037 SRR17174840 Tomato     
 6 Tomato_5    SRX13358036 SRR17174841 Tomato     
 7 Tomato_6    SRX13358035 SRR17174842 Tomato     
 8 NC_1        SRX13358032 SRR17174845 NC         
 9 NC_2        SRX13358026 SRR17174851 NC         
10 NC_3        SRX13358024 SRR17174853 NC         
11 NC_4        SRX13358022 SRR17174855 NC         
12 NC_5        SRX13358630 SRR17174856 NC         
13 NC_6        SRX13358517 SRR17174969 NC         </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-table summarize <span class="dt">\</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-table</span> table-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> table-dada2-tomato-16s.qzv <span class="dt">\</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--m-sample-metadata-file</span> Experiment_Design.tsv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-table tabulate-seqs <span class="dt">\</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-data</span> rep-seqs-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> rep-seqs-dada2-tomato-16s.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>El archivo <code>table-dada2-tomato-16s.qza</code> contiene la matriz de cuentas; en el apéndice <a href="#abrir-archivos-de-qiime2-en-r">Abrir archivos de Qiime2 en R</a> podrás ver a detlla como trabajar con los archivos <code>.qza</code> en <code>R</code>; los primeros 10 ASV se ven del siguiente modo:</p>
<div class="cell">
<div class="cell-output-display">
<div id="titiguxmwf" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#titiguxmwf table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#titiguxmwf thead, #titiguxmwf tbody, #titiguxmwf tfoot, #titiguxmwf tr, #titiguxmwf td, #titiguxmwf th {
  border-style: none;
}

#titiguxmwf p {
  margin: 0;
  padding: 0;
}

#titiguxmwf .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#titiguxmwf .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#titiguxmwf .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#titiguxmwf .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#titiguxmwf .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#titiguxmwf .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#titiguxmwf .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#titiguxmwf .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#titiguxmwf .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#titiguxmwf .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#titiguxmwf .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#titiguxmwf .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#titiguxmwf .gt_spanner_row {
  border-bottom-style: hidden;
}

#titiguxmwf .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#titiguxmwf .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#titiguxmwf .gt_from_md > :first-child {
  margin-top: 0;
}

#titiguxmwf .gt_from_md > :last-child {
  margin-bottom: 0;
}

#titiguxmwf .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#titiguxmwf .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#titiguxmwf .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#titiguxmwf .gt_row_group_first td {
  border-top-width: 2px;
}

#titiguxmwf .gt_row_group_first th {
  border-top-width: 2px;
}

#titiguxmwf .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#titiguxmwf .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#titiguxmwf .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#titiguxmwf .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#titiguxmwf .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#titiguxmwf .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#titiguxmwf .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#titiguxmwf .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#titiguxmwf .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#titiguxmwf .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#titiguxmwf .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#titiguxmwf .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#titiguxmwf .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#titiguxmwf .gt_left {
  text-align: left;
}

#titiguxmwf .gt_center {
  text-align: center;
}

#titiguxmwf .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#titiguxmwf .gt_font_normal {
  font-weight: normal;
}

#titiguxmwf .gt_font_bold {
  font-weight: bold;
}

#titiguxmwf .gt_font_italic {
  font-style: italic;
}

#titiguxmwf .gt_super {
  font-size: 65%;
}

#titiguxmwf .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#titiguxmwf .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#titiguxmwf .gt_indent_1 {
  text-indent: 5px;
}

#titiguxmwf .gt_indent_2 {
  text-indent: 10px;
}

#titiguxmwf .gt_indent_3 {
  text-indent: 15px;
}

#titiguxmwf .gt_indent_4 {
  text-indent: 20px;
}

#titiguxmwf .gt_indent_5 {
  text-indent: 25px;
}

#titiguxmwf .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#titiguxmwf div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="ASV" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ASV</th>
<th id="NC_1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">NC_1</th>
<th id="NC_2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">NC_2</th>
<th id="NC_3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">NC_3</th>
<th id="NC_4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">NC_4</th>
<th id="NC_5" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">NC_5</th>
<th id="NC_6" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">NC_6</th>
<th id="Tomato_1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Tomato_1</th>
<th id="Tomato_2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Tomato_2</th>
<th id="Tomato_3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Tomato_3</th>
<th id="Tomato_4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Tomato_4</th>
<th id="Tomato_5" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Tomato_5</th>
<th id="Tomato_6" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Tomato_6</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="ASV">9002efc8d0aae0e00570a0ef34a8f2f4</td>
<td class="gt_row gt_right" headers="NC_1">0</td>
<td class="gt_row gt_right" headers="NC_2">0</td>
<td class="gt_row gt_right" headers="NC_3">0</td>
<td class="gt_row gt_right" headers="NC_4">0</td>
<td class="gt_row gt_right" headers="NC_5">0</td>
<td class="gt_row gt_right" headers="NC_6">0</td>
<td class="gt_row gt_right" headers="Tomato_1">1280</td>
<td class="gt_row gt_right" headers="Tomato_2">1170</td>
<td class="gt_row gt_right" headers="Tomato_3">3153</td>
<td class="gt_row gt_right" headers="Tomato_4">1873</td>
<td class="gt_row gt_right" headers="Tomato_5">795</td>
<td class="gt_row gt_right" headers="Tomato_6">772</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ASV">41ae7e08f17de260576e380c2f703a0f</td>
<td class="gt_row gt_right" headers="NC_1">20</td>
<td class="gt_row gt_right" headers="NC_2">1360</td>
<td class="gt_row gt_right" headers="NC_3">537</td>
<td class="gt_row gt_right" headers="NC_4">161</td>
<td class="gt_row gt_right" headers="NC_5">1272</td>
<td class="gt_row gt_right" headers="NC_6">3080</td>
<td class="gt_row gt_right" headers="Tomato_1">36</td>
<td class="gt_row gt_right" headers="Tomato_2">18</td>
<td class="gt_row gt_right" headers="Tomato_3">82</td>
<td class="gt_row gt_right" headers="Tomato_4">116</td>
<td class="gt_row gt_right" headers="Tomato_5">21</td>
<td class="gt_row gt_right" headers="Tomato_6">66</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ASV">0be25e4c445f978d165c30f36c9b41ae</td>
<td class="gt_row gt_right" headers="NC_1">0</td>
<td class="gt_row gt_right" headers="NC_2">0</td>
<td class="gt_row gt_right" headers="NC_3">0</td>
<td class="gt_row gt_right" headers="NC_4">0</td>
<td class="gt_row gt_right" headers="NC_5">0</td>
<td class="gt_row gt_right" headers="NC_6">0</td>
<td class="gt_row gt_right" headers="Tomato_1">807</td>
<td class="gt_row gt_right" headers="Tomato_2">1201</td>
<td class="gt_row gt_right" headers="Tomato_3">682</td>
<td class="gt_row gt_right" headers="Tomato_4">393</td>
<td class="gt_row gt_right" headers="Tomato_5">537</td>
<td class="gt_row gt_right" headers="Tomato_6">955</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ASV">d1316b26e5350da36e6456499ecdfe18</td>
<td class="gt_row gt_right" headers="NC_1">72</td>
<td class="gt_row gt_right" headers="NC_2">0</td>
<td class="gt_row gt_right" headers="NC_3">326</td>
<td class="gt_row gt_right" headers="NC_4">102</td>
<td class="gt_row gt_right" headers="NC_5">283</td>
<td class="gt_row gt_right" headers="NC_6">54</td>
<td class="gt_row gt_right" headers="Tomato_1">401</td>
<td class="gt_row gt_right" headers="Tomato_2">694</td>
<td class="gt_row gt_right" headers="Tomato_3">498</td>
<td class="gt_row gt_right" headers="Tomato_4">355</td>
<td class="gt_row gt_right" headers="Tomato_5">426</td>
<td class="gt_row gt_right" headers="Tomato_6">819</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ASV">1624f6ce2f3a06b5b884c321e3d24587</td>
<td class="gt_row gt_right" headers="NC_1">0</td>
<td class="gt_row gt_right" headers="NC_2">0</td>
<td class="gt_row gt_right" headers="NC_3">0</td>
<td class="gt_row gt_right" headers="NC_4">0</td>
<td class="gt_row gt_right" headers="NC_5">0</td>
<td class="gt_row gt_right" headers="NC_6">0</td>
<td class="gt_row gt_right" headers="Tomato_1">543</td>
<td class="gt_row gt_right" headers="Tomato_2">537</td>
<td class="gt_row gt_right" headers="Tomato_3">1332</td>
<td class="gt_row gt_right" headers="Tomato_4">929</td>
<td class="gt_row gt_right" headers="Tomato_5">358</td>
<td class="gt_row gt_right" headers="Tomato_6">314</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ASV">684433f1b511ba86c277251510fce2d7</td>
<td class="gt_row gt_right" headers="NC_1">16</td>
<td class="gt_row gt_right" headers="NC_2">0</td>
<td class="gt_row gt_right" headers="NC_3">9</td>
<td class="gt_row gt_right" headers="NC_4">49</td>
<td class="gt_row gt_right" headers="NC_5">128</td>
<td class="gt_row gt_right" headers="NC_6">11</td>
<td class="gt_row gt_right" headers="Tomato_1">413</td>
<td class="gt_row gt_right" headers="Tomato_2">312</td>
<td class="gt_row gt_right" headers="Tomato_3">661</td>
<td class="gt_row gt_right" headers="Tomato_4">266</td>
<td class="gt_row gt_right" headers="Tomato_5">529</td>
<td class="gt_row gt_right" headers="Tomato_6">785</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ASV">6e21d8726024d1197fad8919bd53ed0a</td>
<td class="gt_row gt_right" headers="NC_1">299</td>
<td class="gt_row gt_right" headers="NC_2">196</td>
<td class="gt_row gt_right" headers="NC_3">262</td>
<td class="gt_row gt_right" headers="NC_4">337</td>
<td class="gt_row gt_right" headers="NC_5">561</td>
<td class="gt_row gt_right" headers="NC_6">283</td>
<td class="gt_row gt_right" headers="Tomato_1">109</td>
<td class="gt_row gt_right" headers="Tomato_2">126</td>
<td class="gt_row gt_right" headers="Tomato_3">187</td>
<td class="gt_row gt_right" headers="Tomato_4">153</td>
<td class="gt_row gt_right" headers="Tomato_5">181</td>
<td class="gt_row gt_right" headers="Tomato_6">68</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ASV">fd63b583680917403001aa46a7cc26ca</td>
<td class="gt_row gt_right" headers="NC_1">6</td>
<td class="gt_row gt_right" headers="NC_2">348</td>
<td class="gt_row gt_right" headers="NC_3">19</td>
<td class="gt_row gt_right" headers="NC_4">25</td>
<td class="gt_row gt_right" headers="NC_5">16</td>
<td class="gt_row gt_right" headers="NC_6">1365</td>
<td class="gt_row gt_right" headers="Tomato_1">24</td>
<td class="gt_row gt_right" headers="Tomato_2">0</td>
<td class="gt_row gt_right" headers="Tomato_3">0</td>
<td class="gt_row gt_right" headers="Tomato_4">55</td>
<td class="gt_row gt_right" headers="Tomato_5">14</td>
<td class="gt_row gt_right" headers="Tomato_6">6</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ASV">bcc71cc1efae148c0f7572338d3e2f86</td>
<td class="gt_row gt_right" headers="NC_1">0</td>
<td class="gt_row gt_right" headers="NC_2">0</td>
<td class="gt_row gt_right" headers="NC_3">0</td>
<td class="gt_row gt_right" headers="NC_4">0</td>
<td class="gt_row gt_right" headers="NC_5">0</td>
<td class="gt_row gt_right" headers="NC_6">0</td>
<td class="gt_row gt_right" headers="Tomato_1">663</td>
<td class="gt_row gt_right" headers="Tomato_2">101</td>
<td class="gt_row gt_right" headers="Tomato_3">769</td>
<td class="gt_row gt_right" headers="Tomato_4">131</td>
<td class="gt_row gt_right" headers="Tomato_5">85</td>
<td class="gt_row gt_right" headers="Tomato_6">67</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ASV">febb741f9cd062f7e0471bd9ad45447f</td>
<td class="gt_row gt_right" headers="NC_1">4</td>
<td class="gt_row gt_right" headers="NC_2">0</td>
<td class="gt_row gt_right" headers="NC_3">34</td>
<td class="gt_row gt_right" headers="NC_4">23</td>
<td class="gt_row gt_right" headers="NC_5">0</td>
<td class="gt_row gt_right" headers="NC_6">1642</td>
<td class="gt_row gt_right" headers="Tomato_1">0</td>
<td class="gt_row gt_right" headers="Tomato_2">0</td>
<td class="gt_row gt_right" headers="Tomato_3">0</td>
<td class="gt_row gt_right" headers="Tomato_4">0</td>
<td class="gt_row gt_right" headers="Tomato_5">0</td>
<td class="gt_row gt_right" headers="Tomato_6">0</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<ul>
<li><p><strong>El comando <code>feature-table summarize</code></strong></p>
<p>Este comando te ayuda a ver:</p>
<ul>
<li><p><strong>Cuántas secuencias hay en cada muestra</strong>: por ejemplo, cuántas secuencias tiene cada una de tus muestras biológicas.</p></li>
<li><p><strong>Cuántas secuencias hay por característica</strong>: las características pueden ser cosas como tipos de bacterias o genes que encontraste, en caso de agregar metadatos.</p></li>
<li><p><strong>Gráficos (histogramas)</strong>: estos te muestran cómo se distribuyen las secuencias.</p></li>
<li><p><strong>Estadísticas básicas</strong>: algunos números que resumen tus datos de forma general, para que tengas una idea rápida de lo que tienes.</p></li>
</ul></li>
<li><p><strong>El comando <code>feature-table tabulate-seqs</code></strong></p>
<p>Este comando es útil porque:</p>
<ul>
<li><p>Te da una lista que conecta cada <strong>ID de característica</strong> (un código único para cada característica) con la secuencia de ADN que representa.</p></li>
<li><p>Además, te ofrece <strong>enlaces</strong> para buscar esas secuencias en una base de datos grande llamada <strong>NCBI nt</strong>.</p></li>
</ul></li>
</ul>
</section>
<section id="medidas-de-diversidad" class="level1">
<h1>Medidas de diversidad</h1>
<section id="generación-de-un-árbol-filogenético" class="level3">
<h3 class="anchored" data-anchor-id="generación-de-un-árbol-filogenético">Generación de un árbol filogenético</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> phylogeny align-to-tree-mafft-fasttree <span class="dt">\</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-sequences</span> rep-seqs-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-alignment</span> aligned-rep-seqs-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-masked-alignment</span> masked-aligned-rep-seqs-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-tree</span> unrooted-tree.qza <span class="dt">\</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-rooted-tree</span> rooted-tree-dada2-tomato-16s.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="métricas-general-de-filogenética" class="level3">
<h3 class="anchored" data-anchor-id="métricas-general-de-filogenética">Métricas general de filogenética</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity core-metrics-phylogenetic <span class="dt">\</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>--i-phylogeny rooted-tree-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>--i-table table-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>--p-sampling-depth 16890 <span class="dt">\</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>--m-metadata-file Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>--output-dir core-metrics-results</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="alfa-diversidad-α-diversidad" class="level2">
<h2 class="anchored" data-anchor-id="alfa-diversidad-α-diversidad">Alfa diversidad (α-diversidad)</h2>
<p>La <strong>alfa diversidad</strong> es la diversidad media de especies microbianas en un lugar a escala local. Representa la riqueza de especies y su abundancia relativa dentro de una sola muestra o hábitat. Es uno de los componentes fundamentales de la diversidad biológica junto con la diversidad beta y gamma.</p>
<section id="características-principales" class="level4">
<h4 class="anchored" data-anchor-id="características-principales">Características principales:</h4>
<ul>
<li>Mide la diversidad dentro de una muestra individual</li>
<li>Considera tanto el número de especies como su abundancia relativa</li>
<li>Es independiente de la escala espacial específica</li>
<li>Fundamental para caracterizar comunidades microbianas</li>
</ul>
</section>
<section id="simpson-index" class="level3">
<h3 class="anchored" data-anchor-id="simpson-index">Simpson Index</h3>
<p>El <strong>índice de Simpson</strong> mide la probabilidad de que dos individuos tomados al azar de una muestra pertenezcan a la misma especie. Es sensible a la dominancia de especies abundantes.</p>
</section>
<section id="fórmula-del-índice-de-simpson" class="level3">
<h3 class="anchored" data-anchor-id="fórmula-del-índice-de-simpson">Fórmula del Índice de Simpson:</h3>
<p><span class="math display">\[D = 1 - \sum_{i=1}^{S} p_i^2\]</span></p>
<p>donde:</p>
<ul>
<li><p><span class="math inline">\(D\)</span> = índice de Simpson (diversidad)</p></li>
<li><p><span class="math inline">\(S\)</span> = número de especies (riqueza)</p></li>
<li><p><span class="math inline">\(p_i\)</span> = proporción de la especie <span class="math inline">\(i\)</span> en la muestra: $p_i = ra</p></li>
</ul>
<p>c{n_i}{N}$</p>
<p>- <span class="math inline">\(n_i\)</span> = número de individuos de la especie <span class="math inline">\(i\)</span> - <span class="math inline">\(N\)</span> = número total de individuos</p>
<p>Algunas variantes:</p>
<p>- <strong>Índice de Simpson original</strong>: <span class="math inline">\(\lambda = \sum p_i^2\)</span></p>
<p>- <strong>Diversidad de Simpson</strong>: <span class="math inline">\(1/\lambda\)</span></p>
</section>
<section id="interpretación" class="level3">
<h3 class="anchored" data-anchor-id="interpretación">Interpretación:</h3>
<ul>
<li><span class="math inline">\(D\)</span> cercano a 1 indica alta diversidad (baja dominancia)</li>
<li><span class="math inline">\(D\)</span> cercano a 0 indica baja diversidad (alta dominancia)</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha <span class="dt">\</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-table</span> table-dada2-tomato-16s.qza <span class="dt">\ </span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--p-metric</span> simpson <span class="dt">\</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-alpha-diversity</span> simpson_diversity.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-group-significance <span class="dt">\</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-alpha-diversity</span> simpson_diversity.qza <span class="dt">\ </span>          </span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--m-metadata-file</span> Experiment_Design.tsv <span class="dt">\ </span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--o-visualization</span> core-metrics-results/simpson-group-significance.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view core-metrics-results/simpson-group-significance.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/simpson_index.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="mflhzwvwjo" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#mflhzwvwjo table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#mflhzwvwjo thead, #mflhzwvwjo tbody, #mflhzwvwjo tfoot, #mflhzwvwjo tr, #mflhzwvwjo td, #mflhzwvwjo th {
  border-style: none;
}

#mflhzwvwjo p {
  margin: 0;
  padding: 0;
}

#mflhzwvwjo .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#mflhzwvwjo .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#mflhzwvwjo .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#mflhzwvwjo .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#mflhzwvwjo .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#mflhzwvwjo .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#mflhzwvwjo .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#mflhzwvwjo .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#mflhzwvwjo .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#mflhzwvwjo .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#mflhzwvwjo .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#mflhzwvwjo .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#mflhzwvwjo .gt_spanner_row {
  border-bottom-style: hidden;
}

#mflhzwvwjo .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#mflhzwvwjo .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#mflhzwvwjo .gt_from_md > :first-child {
  margin-top: 0;
}

#mflhzwvwjo .gt_from_md > :last-child {
  margin-bottom: 0;
}

#mflhzwvwjo .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#mflhzwvwjo .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#mflhzwvwjo .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#mflhzwvwjo .gt_row_group_first td {
  border-top-width: 2px;
}

#mflhzwvwjo .gt_row_group_first th {
  border-top-width: 2px;
}

#mflhzwvwjo .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#mflhzwvwjo .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#mflhzwvwjo .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#mflhzwvwjo .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#mflhzwvwjo .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#mflhzwvwjo .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#mflhzwvwjo .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#mflhzwvwjo .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#mflhzwvwjo .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#mflhzwvwjo .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#mflhzwvwjo .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#mflhzwvwjo .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#mflhzwvwjo .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#mflhzwvwjo .gt_left {
  text-align: left;
}

#mflhzwvwjo .gt_center {
  text-align: center;
}

#mflhzwvwjo .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#mflhzwvwjo .gt_font_normal {
  font-weight: normal;
}

#mflhzwvwjo .gt_font_bold {
  font-weight: bold;
}

#mflhzwvwjo .gt_font_italic {
  font-style: italic;
}

#mflhzwvwjo .gt_super {
  font-size: 65%;
}

#mflhzwvwjo .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#mflhzwvwjo .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#mflhzwvwjo .gt_indent_1 {
  text-indent: 5px;
}

#mflhzwvwjo .gt_indent_2 {
  text-indent: 10px;
}

#mflhzwvwjo .gt_indent_3 {
  text-indent: 15px;
}

#mflhzwvwjo .gt_indent_4 {
  text-indent: 20px;
}

#mflhzwvwjo .gt_indent_5 {
  text-indent: 25px;
}

#mflhzwvwjo .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#mflhzwvwjo div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="Group-1" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 1</th>
<th id="Group-2" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 2</th>
<th id="H" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">H</th>
<th id="p-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">p-value</th>
<th id="q-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">q-value</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Group 1">NC (n=6)</td>
<td class="gt_row gt_left" headers="Group 2">Tomato (n=6)</td>
<td class="gt_row gt_right" headers="H">3.692308</td>
<td class="gt_row gt_right" headers="p-value">0.05466394</td>
<td class="gt_row gt_right" headers="q-value">0.05466394</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="shannon" class="level3">
<h3 class="anchored" data-anchor-id="shannon">Shannon</h3>
<p>El <strong>índice de Shannon</strong> (Shannon-Weaver o Shannon-Wiener) mide la biodiversidad específica considerando tanto la riqueza de especies como su abundancia relativa. Está basado en la teoría de la información y mide la incertidumbre para predecir la especie de un individuo tomado al azar.</p>
<section id="fórmula-del-índice-de-shannon" class="level4">
<h4 class="anchored" data-anchor-id="fórmula-del-índice-de-shannon">Fórmula del índice de Shannon:</h4>
<p><span class="math display">\[H' = -\sum_{i=1}^{S} p_i \ln p_i\]</span></p>
<p>donde:</p>
<p>- <span class="math inline">\(H'\)</span> = índice de Shannon</p>
<p>- <span class="math inline">\(S\)</span> = número de especies (riqueza de especies)</p>
<p>- <span class="math inline">\(p_i\)</span> = proporción de individuos de la especie <span class="math inline">\(i\)</span> respecto al total: <span class="math inline">\(p_i = \frac{n_i}{N}\)</span></p>
<p>- <span class="math inline">\(n_i\)</span> = número de individuos de la especie <span class="math inline">\(i\)</span></p>
<p>- <span class="math inline">\(N\)</span> = número total de individuos de todas las especies</p>
<p>- <span class="math inline">\(\ln\)</span> = logaritmo natural</p>
</section>
<section id="interpretación-1" class="level4">
<h4 class="anchored" data-anchor-id="interpretación-1">Interpretación:</h4>
<ul>
<li>Valores entre 0.5 y 5 en ecosistemas naturales</li>
<li>Valores &lt; 2: baja diversidad</li>
<li>Valores &gt; 3: alta diversidad</li>
<li>Valores más altos indican mayor diversidad y equitabilidad</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha <span class="dt">\</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-table</span> table-dada2-tomato-16s.qza <span class="dt">\ </span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--p-metric</span> shannon <span class="dt">\</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-alpha-diversity</span> shannon_diversity.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-group-significance <span class="dt">\</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-alpha-diversity</span> shanon_diversity.qza <span class="dt">\ </span>          </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--m-metadata-file</span> Experiment_Design.tsv <span class="dt">\ </span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--o-visualization</span> core-metrics-results/shannon-group-significance.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view core-metrics-results/shannon-group-significance.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/shannon_ex1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/shannon_ex2.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="otras-métricas" class="level3">
<h3 class="anchored" data-anchor-id="otras-métricas">Otras métricas</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-group-significance <span class="dt">\</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>--i-alpha-diversity core-metrics-results/faith_pd_vector.qza <span class="dt">\</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>--m-metadata-file Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>--o-visualization core-metrics-results/faith-pd-group-significance.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-group-significance <span class="dt">\</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>--i-alpha-diversity core-metrics-results/evenness_vector.qza <span class="dt">\</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>--m-metadata-file Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>--o-visualization core-metrics-results/evenness-group-significance.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Los siguientes gráficos y tablas de evenness y faith se obtuvieron a partir de los archivos <code>.qzv</code> generados, asi como las tablas de significancia <code>kruskal-wallis-pairwise</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view evenness-group-significance.qzv</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view faith-pd-group-significance.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="eveness-equitabilidad" class="level3">
<h3 class="anchored" data-anchor-id="eveness-equitabilidad">Eveness (equitabilidad)</h3>
<p>La <strong>Evenness</strong> o <strong>equitabilidad</strong> mide qué tan uniformemente están distribuidas las abundancias de las especies en una comunidad. Una alta equitabilidad indica que las especies tienen abundancias similares.</p>
<section id="índice-de-equitabilidad-de-pielou" class="level4">
<h4 class="anchored" data-anchor-id="índice-de-equitabilidad-de-pielou">Índice de Equitabilidad de Pielou:</h4>
<p><span class="math display">\[J' = \frac{H'}{H'_{\max}} = \frac{H'}{\ln S}\]</span></p>
<p>donde:</p>
<p>- <span class="math inline">\(J'\)</span> = índice de equitabilidad de Pielou</p>
<p>- <span class="math inline">\(H'\)</span> = índice de Shannon observado</p>
<p>- <span class="math inline">\(H'_{\max}\)</span> = diversidad máxima posible = <span class="math inline">\(\ln S\)</span></p>
<p>- <span class="math inline">\(S\)</span> = número de especies</p>
</section>
<section id="interpretación-2" class="level4">
<h4 class="anchored" data-anchor-id="interpretación-2">Interpretación:</h4>
<ul>
<li>Valores entre 0 y 1</li>
<li>0 = máxima inequidad (una especie domina completamente)</li>
<li>1 = máxima equidad (todas las especies igualmente abundantes)</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/alpha-compare.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="aplvjbwhaw" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#aplvjbwhaw table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#aplvjbwhaw thead, #aplvjbwhaw tbody, #aplvjbwhaw tfoot, #aplvjbwhaw tr, #aplvjbwhaw td, #aplvjbwhaw th {
  border-style: none;
}

#aplvjbwhaw p {
  margin: 0;
  padding: 0;
}

#aplvjbwhaw .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#aplvjbwhaw .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#aplvjbwhaw .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#aplvjbwhaw .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#aplvjbwhaw .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#aplvjbwhaw .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aplvjbwhaw .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#aplvjbwhaw .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#aplvjbwhaw .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#aplvjbwhaw .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#aplvjbwhaw .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#aplvjbwhaw .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#aplvjbwhaw .gt_spanner_row {
  border-bottom-style: hidden;
}

#aplvjbwhaw .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#aplvjbwhaw .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#aplvjbwhaw .gt_from_md > :first-child {
  margin-top: 0;
}

#aplvjbwhaw .gt_from_md > :last-child {
  margin-bottom: 0;
}

#aplvjbwhaw .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#aplvjbwhaw .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#aplvjbwhaw .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#aplvjbwhaw .gt_row_group_first td {
  border-top-width: 2px;
}

#aplvjbwhaw .gt_row_group_first th {
  border-top-width: 2px;
}

#aplvjbwhaw .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#aplvjbwhaw .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#aplvjbwhaw .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#aplvjbwhaw .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aplvjbwhaw .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#aplvjbwhaw .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#aplvjbwhaw .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#aplvjbwhaw .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#aplvjbwhaw .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aplvjbwhaw .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#aplvjbwhaw .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#aplvjbwhaw .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#aplvjbwhaw .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#aplvjbwhaw .gt_left {
  text-align: left;
}

#aplvjbwhaw .gt_center {
  text-align: center;
}

#aplvjbwhaw .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#aplvjbwhaw .gt_font_normal {
  font-weight: normal;
}

#aplvjbwhaw .gt_font_bold {
  font-weight: bold;
}

#aplvjbwhaw .gt_font_italic {
  font-style: italic;
}

#aplvjbwhaw .gt_super {
  font-size: 65%;
}

#aplvjbwhaw .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#aplvjbwhaw .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#aplvjbwhaw .gt_indent_1 {
  text-indent: 5px;
}

#aplvjbwhaw .gt_indent_2 {
  text-indent: 10px;
}

#aplvjbwhaw .gt_indent_3 {
  text-indent: 15px;
}

#aplvjbwhaw .gt_indent_4 {
  text-indent: 20px;
}

#aplvjbwhaw .gt_indent_5 {
  text-indent: 25px;
}

#aplvjbwhaw .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#aplvjbwhaw div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="Group-1" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 1</th>
<th id="Group-2" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 2</th>
<th id="H" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">H</th>
<th id="p-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">p-value</th>
<th id="q-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">q-value</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Group 1">NC (n=6)</td>
<td class="gt_row gt_left" headers="Group 2">Tomato (n=6)</td>
<td class="gt_row gt_right" headers="H">3.692308</td>
<td class="gt_row gt_right" headers="p-value">0.05466394</td>
<td class="gt_row gt_right" headers="q-value">0.05466394</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
</section>
<section id="faiths-phylogenetic-diversity-pd" class="level3">
<h3 class="anchored" data-anchor-id="faiths-phylogenetic-diversity-pd">Faith’s Phylogenetic Diversity (PD)</h3>
<p><strong>Faith’s Phylogenetic Diversity (PD)</strong> es un índice que incorpora información filogenética para medir la diversidad evolutiva de una comunidad. No solo considera el número de especies sino también sus relaciones evolutivas.</p>
<section id="fórmula-de-faith-pd" class="level4">
<h4 class="anchored" data-anchor-id="fórmula-de-faith-pd">Fórmula de Faith PD:</h4>
<p><span class="math display">\[PD = \sum_{i=1}^{N} L_i\]</span></p>
<p>donde:</p>
<p>- <span class="math inline">\(PD\)</span> = diversidad filogenética de Faith</p>
<p>- <span class="math inline">\(N\)</span> = número de ramas del árbol filogenético que conectan las especies presentes</p>
<p>- <span class="math inline">\(L_i\)</span> = longitud de la rama <span class="math inline">\(i\)</span> en el árbol filogenético</p>
<p>- Las ramas se suman desde las especies hasta la raíz del árbol</p>
</section>
<section id="características-principales-1" class="level4">
<h4 class="anchored" data-anchor-id="características-principales-1">Características principales:</h4>
<ul>
<li>Incorpora historia evolutiva</li>
<li>Útil para prioridades de conservación</li>
<li>Medida en unidades de tiempo evolutivo</li>
<li>Mayor PD indica mayor diversidad evolutiva</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/alpha-compare_faith.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="jigbzanwmp" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#jigbzanwmp table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#jigbzanwmp thead, #jigbzanwmp tbody, #jigbzanwmp tfoot, #jigbzanwmp tr, #jigbzanwmp td, #jigbzanwmp th {
  border-style: none;
}

#jigbzanwmp p {
  margin: 0;
  padding: 0;
}

#jigbzanwmp .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#jigbzanwmp .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#jigbzanwmp .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#jigbzanwmp .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#jigbzanwmp .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jigbzanwmp .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jigbzanwmp .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jigbzanwmp .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#jigbzanwmp .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#jigbzanwmp .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#jigbzanwmp .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#jigbzanwmp .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#jigbzanwmp .gt_spanner_row {
  border-bottom-style: hidden;
}

#jigbzanwmp .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#jigbzanwmp .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#jigbzanwmp .gt_from_md > :first-child {
  margin-top: 0;
}

#jigbzanwmp .gt_from_md > :last-child {
  margin-bottom: 0;
}

#jigbzanwmp .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#jigbzanwmp .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#jigbzanwmp .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#jigbzanwmp .gt_row_group_first td {
  border-top-width: 2px;
}

#jigbzanwmp .gt_row_group_first th {
  border-top-width: 2px;
}

#jigbzanwmp .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jigbzanwmp .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#jigbzanwmp .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#jigbzanwmp .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jigbzanwmp .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jigbzanwmp .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#jigbzanwmp .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#jigbzanwmp .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#jigbzanwmp .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jigbzanwmp .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jigbzanwmp .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jigbzanwmp .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jigbzanwmp .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jigbzanwmp .gt_left {
  text-align: left;
}

#jigbzanwmp .gt_center {
  text-align: center;
}

#jigbzanwmp .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#jigbzanwmp .gt_font_normal {
  font-weight: normal;
}

#jigbzanwmp .gt_font_bold {
  font-weight: bold;
}

#jigbzanwmp .gt_font_italic {
  font-style: italic;
}

#jigbzanwmp .gt_super {
  font-size: 65%;
}

#jigbzanwmp .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#jigbzanwmp .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#jigbzanwmp .gt_indent_1 {
  text-indent: 5px;
}

#jigbzanwmp .gt_indent_2 {
  text-indent: 10px;
}

#jigbzanwmp .gt_indent_3 {
  text-indent: 15px;
}

#jigbzanwmp .gt_indent_4 {
  text-indent: 20px;
}

#jigbzanwmp .gt_indent_5 {
  text-indent: 25px;
}

#jigbzanwmp .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#jigbzanwmp div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="Group-1" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 1</th>
<th id="Group-2" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 2</th>
<th id="H" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">H</th>
<th id="p-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">p-value</th>
<th id="q-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">q-value</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Group 1">NC (n=6)</td>
<td class="gt_row gt_left" headers="Group 2">Tomato (n=6)</td>
<td class="gt_row gt_right" headers="H">8.307692</td>
<td class="gt_row gt_right" headers="p-value">0.003947752</td>
<td class="gt_row gt_right" headers="q-value">0.003947752</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
</section>
</section>
<section id="curvas-de-rarefacción" class="level2">
<h2 class="anchored" data-anchor-id="curvas-de-rarefacción">Curvas de rarefacción</h2>
<p>La <strong>alfa-rarefacción</strong> es una técnica para evaluar cómo crece la diversidad “intra-muestra” (α-diversidad) a medida que aumentas el número de lecturas secuenciadas. En metagenómica o secuenciación 16S, su objetivo principal es:</p>
<ol type="1">
<li><p><strong>Submuestrear</strong> tu biblioteca de lecturas a distintos “depths” (profundidades de secuenciación): por ejemplo, tomar aleatoriamente 100, 200, …, 4 000 lecturas de cada muestra.</p></li>
<li><p><strong>Calcular un índice de α-diversidad</strong> (observed OTUs, Shannon, Simpson, Faith’s PD, etc.) en cada submuestra.</p></li>
<li><p><strong>Trazar</strong> ese índice frente al número de lecturas: obtienes una curva de rarefacción.</p></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-rarefaction <span class="dt">\</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>--i-table table-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>--i-phylogeny rooted-tree-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>--p-max-depth 20000 <span class="dt">\</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>--m-metadata-file Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>--o-visualization alpha-rarefaction-dada2-tomato-16s.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view alpha-rarefaction-dada2-tomato-16s.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="observed-features" class="level3">
<h3 class="anchored" data-anchor-id="observed-features">Observed Features</h3>
<p>Las <strong>Observed Features</strong> representan simplemente el <strong>número de características taxonómicas diferentes</strong> (especies, géneros, familias, etc.) observadas en una muestra. Es la medida más básica de riqueza de especies.</p>
<section id="características-principales-2" class="level4">
<h4 class="anchored" data-anchor-id="características-principales-2">Características principales:</h4>
<ul>
<li>Medida directa de riqueza taxonómica</li>
<li>No considera abundancia relativa</li>
<li>Sensible al esfuerzo de muestreo</li>
<li>Base para otros índices de diversidad</li>
</ul>
<p>La fórmula es simplemente:</p>
<p><span class="math display">\[S = \text{número total de especies observadas}\]</span></p>
<p>donde:</p>
<p>- <span class="math inline">\(S\)</span> = riqueza de especies observadas</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/features_rarefaction.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="shannon-1" class="level3">
<h3 class="anchored" data-anchor-id="shannon-1">Shannon</h3>
<section id="interpretación-3" class="level4">
<h4 class="anchored" data-anchor-id="interpretación-3">Interpretación:</h4>
<ul>
<li>Valores entre 0.5 y 5 en ecosistemas naturales</li>
<li>Valores &lt; 2: baja diversidad</li>
<li>Valores &gt; 3: alta diversidad</li>
<li>Valores más altos indican mayor diversidad y equitabilidad</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/shannon_rarefaction.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="faith-pd" class="level3">
<h3 class="anchored" data-anchor-id="faith-pd">Faith PD</h3>
<p>La diversidad filogenética de Faith (PD de Faith) es una medida de la biodiversidad que cuantifica la amplitud de la historia evolutiva representada por un conjunto de especies u otras unidades taxonómicas, definida como la suma de las longitudes de las ramas de un árbol filogenético que conectan los miembros del conjunto.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/faith_rarefaction.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="beta-diversidad-β-diversidad" class="level2">
<h2 class="anchored" data-anchor-id="beta-diversidad-β-diversidad">Beta diversidad (β-diversidad)</h2>
<p>La <strong>beta diversidad</strong> es la relación entre la diversidad regional y la diversidad local de especies. Mide el grado de diferenciación o recambio de especies entre diferentes sitios o comunidades. En microbiología, nos permite comparar qué tan similares o diferentes son las comunidades microbianas entre distintas muestras.</p>
<section id="características-principales-3" class="level4">
<h4 class="anchored" data-anchor-id="características-principales-3">Características principales:</h4>
<ul>
<li>Mide la variación en composición entre comunidades</li>
<li>Puede deberse a recambio de especies o anidamiento</li>
<li>Valores altos indican comunidades muy diferentes</li>
<li>Valores bajos indican comunidades similares</li>
</ul>
<p>La <strong>diversidad β verdadera</strong> se calcula como:</p>
<p><span class="math display">\[\beta = \frac{\gamma}{\alpha}\]</span></p>
<p>donde:</p>
<p>- <span class="math inline">\(\beta\)</span> = diversidad beta verdadera (número efectivo de comunidades)</p>
<p>- <span class="math inline">\(\gamma\)</span> = diversidad total del conjunto de datos</p>
<p>- <span class="math inline">\(\alpha\)</span> = diversidad alfa promedio</p>
</section>
<section id="distancia-bray-curtis" class="level3">
<h3 class="anchored" data-anchor-id="distancia-bray-curtis">Distancia Bray Curtis</h3>
<p>La <strong>distancia de Bray-Curtis</strong> es una medida de disimilitud composicional entre dos sitios basada en las abundancias de especies. Es ampliamente utilizada en ecología microbiana para comparar comunidades.</p>
<section id="fórmula-de-bray-curtis" class="level4">
<h4 class="anchored" data-anchor-id="fórmula-de-bray-curtis">Fórmula de Bray-Curtis:</h4>
<p><span class="math display">\[BC_{ij} = 1 - \frac{2C_{ij}}{S_i + S_j}\]</span></p>
<p>o de forma equivalente:</p>
<p><span class="math display">\[BC_{ij} = \frac{\sum_{k=1}^{n} |x_{ik} - x_{jk}|}{\sum_{k=1}^{n} (x_{ik} + x_{jk})}\]</span></p>
<p>donde:</p>
<p>- <span class="math inline">\(BC_{ij}\)</span> = distancia de Bray-Curtis entre sitios <span class="math inline">\(i\)</span> y <span class="math inline">\(j\)</span></p>
<p>- <span class="math inline">\(C_{ij}\)</span> = suma de los valores menores para especies comunes entre ambos sitios</p>
<p>- <span class="math inline">\(S_i\)</span>, <span class="math inline">\(S_j\)</span> = número total de individuos contados en los sitios <span class="math inline">\(i\)</span> y <span class="math inline">\(j\)</span></p>
<p>- <span class="math inline">\(x_{ik}\)</span>, <span class="math inline">\(x_{jk}\)</span> = abundancia de la especie <span class="math inline">\(k\)</span> en los sitios <span class="math inline">\(i\)</span> y <span class="math inline">\(j\)</span></p>
<p>- <span class="math inline">\(n\)</span> = número total de especies</p>
</section>
<section id="interpretación-4" class="level4">
<h4 class="anchored" data-anchor-id="interpretación-4">Interpretación:</h4>
<ul>
<li>Valores entre 0 y 1</li>
<li>0 = sitios idénticos en composición</li>
<li>1 = sitios completamente diferentes</li>
<li>No considera relaciones filogenéticas</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view core-metrics-results/bray_curtis_emperor.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/bray_curtis.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
<section id="permanova" class="level4">
<h4 class="anchored" data-anchor-id="permanova">PERMANOVA</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity beta-group-significance <span class="dt">\</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>--i-distance-matrix core-metrics-results/bray_curtis_distance_matrix.qza <span class="dt">\ </span>    </span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="ex">--m-metadata-file</span> Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>--o-visualization core-metrics-results/curtis-significance.qzv <span class="dt">\</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>--p-pairwise <span class="at">--m-metadata-column</span> Type</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view core-metrics-results/curtis-significance.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/permanova_curtis_boxplot.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/permanova_curtis_table.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="distancia-de-jaccard" class="level3">
<h3 class="anchored" data-anchor-id="distancia-de-jaccard">Distancia de Jaccard</h3>
<p>La <strong>distancia de Jaccard</strong> mide la disimilitud entre dos conjuntos basándose únicamente en presencia/ausencia, sin considerar abundancias. Es útil para datos binarios.</p>
<section id="fórmula-del-índice-de-jaccard" class="level4">
<h4 class="anchored" data-anchor-id="fórmula-del-índice-de-jaccard">Fórmula del índice de Jaccard:</h4>
<p><span class="math display">\[J(A,B) = \frac{|A \cap B|}{|A \cup B|}\]</span></p>
</section>
<section id="fórmula-de-la-distancia-de-jaccard" class="level4">
<h4 class="anchored" data-anchor-id="fórmula-de-la-distancia-de-jaccard">Fórmula de la distancia de Jaccard:</h4>
<p><span class="math display">\[D_J(A,B) = 1 - J(A,B) = 1 - \frac{|A \cap B|}{|A \cup B|}\]</span></p>
<p>o en términos ecológicos:</p>
<p><span class="math display">\[J = \frac{c}{a + b - c}\]</span></p>
<p>donde:</p>
<p>- <span class="math inline">\(J\)</span> = índice de similitud de Jaccard</p>
<p>- <span class="math inline">\(a\)</span> = número de especies presentes en el sitio A</p>
<p>- <span class="math inline">\(b\)</span> = número de especies presentes en el sitio B<br>
- <span class="math inline">\(c\)</span> = número de especies presentes en ambos sitios A y B</p>
<p>- <span class="math inline">\(D_J\)</span> = distancia de Jaccard</p>
</section>
<section id="interpretación-5" class="level4">
<h4 class="anchored" data-anchor-id="interpretación-5">Interpretación:</h4>
<ul>
<li>Valores entre 0 y 1</li>
<li>0 = sitios idénticos (especies presentes son las mismas)</li>
<li>1 = sitios completamente diferentes (no comparten especies)</li>
<li>No considera abundancias, solo presencia/ausencia</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view core-metrics-results/jaccard_emperor.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/distancia_jaccard.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
<section id="permanova-1" class="level4">
<h4 class="anchored" data-anchor-id="permanova-1">PERMANOVA</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity beta-group-significance <span class="dt">\</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>--i-distance-matrix core-metrics-results/jaccard_distance_matrix.qza <span class="dt">\ </span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="ex">--m-metadata-file</span> Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>--o-visualization core-metrics-results/jaccard-significance.qzv <span class="dt">\</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>--p-pairwise <span class="at">--m-metadata-column</span> Type</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view core-metrics-results/jaccard-significance.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/permanova_jaccard_boxplot.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/permanova_jaccard_table.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="unifrac" class="level3">
<h3 class="anchored" data-anchor-id="unifrac">UniFrac</h3>
<p><strong>UniFrac</strong> es una métrica de distancia filogenética utilizada para comparar comunidades biológicas. Incorpora información sobre las relaciones filogenéticas entre los organismos observados, diferenciándose de medidas como Bray-Curtis.</p>
<section id="unifrac-no-ponderado-unweighted" class="level4">
<h4 class="anchored" data-anchor-id="unifrac-no-ponderado-unweighted">UniFrac No Ponderado (Unweighted):</h4>
<p><span class="math display">\[U(A,B) = \frac{\sum_{i=1}^{N} l_i \times |A_i - B_i|}{\sum_{i=1}^{N} l_i \times \max(A_i, B_i)}\]</span></p>
<p>donde:</p>
<p>- <span class="math inline">\(U(A,B)\)</span> = distancia UniFrac no ponderada entre muestras A y B</p>
<p>- <span class="math inline">\(N\)</span> = número de nodos en el árbol filogenético</p>
<p>- <span class="math inline">\(l_i\)</span> = longitud de la rama entre el nodo <span class="math inline">\(i\)</span> y su nodo parental</p>
<p>- <span class="math inline">\(A_i\)</span> = indicador (1 o 0) de presencia de la muestra A en las ramas descendientes del nodo <span class="math inline">\(i\)</span></p>
<p>- <span class="math inline">\(B_i\)</span> = indicador (1 o 0) de presencia de la muestra B en las ramas descendientes del nodo <span class="math inline">\(i\)</span></p>
</section>
<section id="unifrac-ponderado-weighted" class="level4">
<h4 class="anchored" data-anchor-id="unifrac-ponderado-weighted">UniFrac Ponderado (Weighted):</h4>
<p><span class="math display">\[W(A,B) = \sum_{i=1}^{N} l_i \times \left|\frac{A_i}{A_t} - \frac{B_i}{B_t}\right|\]</span></p>
<p>donde:</p>
<p>- <span class="math inline">\(W(A,B)\)</span> = distancia UniFrac ponderada entre muestras A y B</p>
<ul>
<li><span class="math inline">\(A_i\)</span> = número de secuencias de la muestra A en las ramas descendientes del nodo <span class="math inline">\(i\)</span></li>
</ul>
<p>- <span class="math inline">\(A_t\)</span> = número total de secuencias en la muestra A</p>
<p>- <span class="math inline">\(B_i\)</span> = número de secuencias de la muestra B en las ramas descendientes del nodo <span class="math inline">\(i\)</span></p>
<p>- <span class="math inline">\(B_t\)</span> = número total de secuencias en la muestra B</p>
</section>
<section id="interpretación-6" class="level4">
<h4 class="anchored" data-anchor-id="interpretación-6">Interpretación:</h4>
<ul>
<li>Valores entre 0 y 1</li>
<li>0 = comunidades idénticas filogenéticamente</li>
<li>1 = comunidades completamente diferentes filogenéticamente</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity beta-group-significance <span class="dt">\</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>--i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza <span class="dt">\</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>--m-metadata-file Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>--m-metadata-column Type <span class="dt">\</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>--o-visualization core-metrics-results/unweighted-unifrac-body-site-significance.qzv <span class="dt">\</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>--p-pairwise</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="unweighted-unifrac" class="level3">
<h3 class="anchored" data-anchor-id="unweighted-unifrac">Unweighted UniFrac</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view core-metrics-results/unweighted_unifrac_emperor.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/sin_ponderar.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<section id="permanova-2" class="level4">
<h4 class="anchored" data-anchor-id="permanova-2">PERMANOVA</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view core-metrics-results/unweighted-unifrac-body-site-significance.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/permanova_boxplots.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/permanova_table.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
</section>
</section>
</section>
<section id="taxonomía" class="level1">
<h1>Taxonomía</h1>
<p><strong>NOTA</strong> ⚠️<strong>:</strong> si no cuentas con un modelo entrenado para clasificar las secuencias, debes checar en al Apéndice <a href="#entrenamiento-base-de-datos-silva">Entrenamiento base de datos SILVA</a>.</p>
<section id="diversidadabundancía" class="level2">
<h2 class="anchored" data-anchor-id="diversidadabundancía">Diversidad/abundancía</h2>
<p>Este comando asigna taxonomía a las secuencias representativas (como ASVs u OTUs) obtenidas de un experimento de secuenciación 16S. Utiliza un clasificador preentrenado basado en machine learning (Naive Bayes) para identificar los taxones a los que pertenecen las secuencias. Este paso es esencial para determinar qué microorganismos están presentes en las muestras según sus secuencias.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier classify-sklearn <span class="dt">\</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-classifier</span> silva-138.2-ssu-nr99-515f-806r-classifier.qza <span class="dt">\</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-reads</span> rep-seqs-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-classification</span> taxonomy.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>El siguiente comando es útil para revisar las asignaciones taxonómicas y explorar la composición de las secuencias de forma interactiva.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> metadata tabulate <span class="dt">\</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--m-input-file</span> taxonomy.qza <span class="dt">\</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> taxonomy.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Este gráfico facilita el análisis visual de cómo varía la microbiota entre diferentes condiciones del experimento.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> taxa barplot <span class="dt">\</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-table</span> table-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-taxonomy</span> taxonomy.qza <span class="dt">\</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> taxa-bar-plots.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view taxa-bar-plots.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/rect469.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
<section id="pruebas-de-abundancia-diferencial-con-ancom-bc" class="level2">
<h2 class="anchored" data-anchor-id="pruebas-de-abundancia-diferencial-con-ancom-bc">Pruebas de abundancia diferencial con ANCOM-BC</h2>
<p><strong>ANCOM-BC</strong> (Analysis of Composition of Microbiomes with Bias Correction) es un método para detectar taxones con abundancia diferencial entre grupos, corregir sesgos de muestreo y normalización.</p>
<section id="pasos-principales-de-ancom-bc" class="level3">
<h3 class="anchored" data-anchor-id="pasos-principales-de-ancom-bc">Pasos Principales de ANCOM-BC:</h3>
<ol type="1">
<li><strong>Modelado de cuentas</strong>: Ajusta modelos lineales generalizados a las cuentas de taxones con corrección de tamaño de muestra.</li>
<li><strong>Estimación de log-fold changes</strong>: Calcula la diferencia de abundancia en escala logarítmica entre condiciones.</li>
<li><strong>Corrección de sesgos</strong>: Ajusta estimaciones para minimizar sesgos debidos a diferencias en la profundidad de secuenciación.</li>
<li><strong>Pruebas estadísticas</strong>: Realiza pruebas Wald para evaluar si los cambios log-fold son significativamente distintos de cero.</li>
</ol>
</section>
<section id="ecuación-general-de-log-fold-change-estimado" class="level3">
<h3 class="anchored" data-anchor-id="ecuación-general-de-log-fold-change-estimado">Ecuación general de log-fold change estimado:</h3>
<p>$$\hat{\eta}_i = \log\!\left(\frac{\hat{\mu}_{i,1}}{\hat{\mu}_{i,0}}\right)$$</p>
<p>donde:</p>
<p>- <span class="math inline">\(\hat{\eta}_i\)</span> = estimación del log-fold change para el taxón <span class="math inline">\(i\)</span></p>
<p>- <span class="math inline">\(\hat{\mu}_{i,1}\)</span> = abundancia media ajustada del taxón <span class="math inline">\(i\)</span> en el grupo 1</p>
<p>- <span class="math inline">\(\hat{\mu}_{i,0}\)</span> = abundancia media ajustada del taxón <span class="math inline">\(i\)</span> en el grupo 0</p>
</section>
<section id="interpretación-de-resultados" class="level3">
<h3 class="anchored" data-anchor-id="interpretación-de-resultados">Interpretación de resultados:</h3>
<ul>
<li><span class="math inline">\(\hat{\eta}_i &gt; 0\)</span> indica mayor abundancia en el grupo 1</li>
<li><span class="math inline">\(\hat{\eta}_i &lt; 0\)</span> indica mayor abundancia en el grupo 0</li>
<li>SE(<span class="math inline">\(\hat{\eta}_i\)</span>) es el error estándar de la estimación</li>
<li><span class="math inline">\(z_i = \hat{\eta}_i / SE(\hat{\eta}_i)\)</span> se utiliza para la prueba Wald</li>
</ul>
<p>ANCOM-BC es especialmente útil en metabarcoding porque:</p>
<ul>
<li><p>Corrige sesgos de muestreo</p></li>
<li><p>Maneja datos escasos y muy dispersos</p></li>
<li><p>Proporciona estimaciones robustas de cambios en abundancia</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> composition ancombc <span class="dt">\</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-table</span> table-dada2-tomato-16s.qza <span class="dt">\ </span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--m-metadata-file</span> Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--p-formula</span> <span class="st">'Type'</span> <span class="dt">\ </span>  </span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--o-differentials</span> ancombc-subject.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> composition da-barplot <span class="dt">\</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-data</span> ancombc-subject.qza <span class="dt">\</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--p-significance-threshold</span> 0.001 <span class="dt">\</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> da-barplot-type.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view da-barplot-type.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/visualization.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>Ahora lo puedes hacer para cada ASV anotado por niveles, en este caso lo haremos con en <code>nivel 3</code> y usaremos de los metadata a <code>Type</code>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> taxa collapse <span class="dt">\</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-table</span> table-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-taxonomy</span> taxonomy.qza <span class="dt">\</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--p-level</span> 3 <span class="dt">\</span></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-collapsed-table</span> tomato-table-l3.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> composition ancombc <span class="dt">\</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-table</span> tomato-table-l3.qza <span class="dt">\ </span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--m-metadata-file</span> Experiment_Design.tsv <span class="dt">\ </span></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--p-formula</span> <span class="st">'Type'</span> <span class="dt">\ </span>  </span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--o-differentials</span> l3-ancombc-type.qza  </span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> composition da-barplot <span class="dt">\</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-data</span> l3-ancombc-type.qza <span class="dt">\ </span> </span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--p-significance-threshold</span> 0.001 <span class="dt">\</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--p-level-delimiter</span> <span class="st">';'</span> <span class="dt">\</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> l3-da-barplot-subject.qzv</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view l3-da-barplot-subject.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/visualization_level3.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>Para el caso de el <code>nivel 6</code> solo habría que cambiar en ese flag:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> taxa collapse <span class="dt">\ </span>                    </span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--i-table</span> table-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-taxonomy</span> taxonomy.qza <span class="dt">\ </span>             </span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--p-level</span> 6 <span class="dt">\ </span>      </span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--o-collapsed-table</span> tomato-table-l6.qza</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> composition ancombc <span class="dt">\ </span>              </span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--i-table</span> tomato-table-l6.qza <span class="dt">\ </span>      </span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--m-metadata-file</span> Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">--p-formula</span> <span class="st">'Type'</span> <span class="dt">\</span></span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-differentials</span> l6-ancombc-type.qza </span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> composition da-barplot <span class="dt">\ </span>           </span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--i-data</span> l6-ancombc-type.qza <span class="dt">\</span></span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">--p-significance-threshold</span> 0.001 <span class="dt">\</span></span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">--p-level-delimiter</span> <span class="st">';'</span> <span class="dt">\</span></span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> l6-da-barplot-subject.qzv</span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view l6-da-barplot-subject.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/visualization_level6.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p>Para explorar clados especificos puede revisar el materíal del Apédice <a href="#boxplots-para-algunos-clados-importantes">Boxplots para algunos clados importantes</a>.</p>
</section>
</section>
</section>
<section id="apéndice" class="level1">
<h1>Apéndice</h1>
<section id="abrir-archivos-de-qiime2-en-r" class="level2">
<h2 class="anchored" data-anchor-id="abrir-archivos-de-qiime2-en-r">Abrir archivos de Qiime2 en R</h2>
<p>Primero instala la paquetería <a href="https://github.com/jbisanz/qiime2R/"><code>qiime2R</code></a></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"devtools"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)){<span class="fu">install.packages</span>(<span class="st">"devtools"</span>)}</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"jbisanz/qiime2R"</span>)</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Esta paquetería nos permite leer los archivos <code>.qza</code> <strong>QIIME Zipped Artifact</strong>.</p>
<p>Por ejemplo, si buscas acceder a la matriz de cuentas de los ASV del archivo <code>table-dada2-tomato-16s.qza</code>, puede utilizar el siguiente código:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qiime2R)</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>df_tmp <span class="ot">&lt;-</span> <span class="fu">read_qza</span>(<span class="st">"../2025_Demo_16S_tomate/table-dada2-tomato-16s.qza"</span>)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>df_counts <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(df_tmp<span class="sc">$</span>data)</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(df_counts, <span class="dv">5</span>)</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                 NC_1 NC_2 NC_3 NC_4 NC_5 NC_6 Tomato_1
9002efc8d0aae0e00570a0ef34a8f2f4    0    0    0    0    0    0     1280
41ae7e08f17de260576e380c2f703a0f   20 1360  537  161 1272 3080       36
0be25e4c445f978d165c30f36c9b41ae    0    0    0    0    0    0      807
d1316b26e5350da36e6456499ecdfe18   72    0  326  102  283   54      401
1624f6ce2f3a06b5b884c321e3d24587    0    0    0    0    0    0      543
                                 Tomato_2 Tomato_3 Tomato_4 Tomato_5 Tomato_6
9002efc8d0aae0e00570a0ef34a8f2f4     1170     3153     1873      795      772
41ae7e08f17de260576e380c2f703a0f       18       82      116       21       66
0be25e4c445f978d165c30f36c9b41ae     1201      682      393      537      955
d1316b26e5350da36e6456499ecdfe18      694      498      355      426      819
1624f6ce2f3a06b5b884c321e3d24587      537     1332      929      358      314</code></pre>
</div>
</div>
</section>
<section id="otras-métricas-de-alfa-diversidad" class="level2">
<h2 class="anchored" data-anchor-id="otras-métricas-de-alfa-diversidad">Otras métricas de Alfa Diversidad</h2>
<p>Para los siguientes gráficos se dercargaron los archivos <code>.csv</code> a partir del archivo <code>.qzv</code>, para posteriormente usar <code>R</code> para gráficar.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view alpha-rarefaction-dada2-tomato-16s.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Se descargaron los siguientes archivos:</p>
<ul>
<li><p><code>observed_features.csv</code></p></li>
<li><p><code>shannon.csv</code></p></li>
<li><p><code>faith_pd.csv</code></p></li>
</ul>
<section id="usando-r-para-gŕaficar" class="level3">
<h3 class="anchored" data-anchor-id="usando-r-para-gŕaficar">Usando R para gŕaficar</h3>
<section id="observed-features-1" class="level4">
<h4 class="anchored" data-anchor-id="observed-features-1">Observed Features</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>obfe <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">"tables/observed_features.csv"</span>,</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col_types =</span> <span class="fu">cols</span>(</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>                 <span class="st">`</span><span class="at">sample-id</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">col_skip</span>(),</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">accession   =</span> <span class="fu">col_skip</span>(),</span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">sra_run     =</span> <span class="fu">col_skip</span>(),</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Type        =</span> <span class="fu">col_factor</span>(),</span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">.default    =</span> <span class="fu">col_double</span>()</span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>               ))</span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a>obfe_long <span class="ot">&lt;-</span> obfe <span class="sc">%&gt;%</span></span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"depth-"</span>), </span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="fu">c</span>(<span class="st">"Depth"</span>, <span class="st">"Iter"</span>),            </span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_pattern =</span> <span class="st">"depth-([0-9]+)_iter-([0-9]+)"</span>,</span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"observed_features"</span>,</span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_drop_na =</span> <span class="cn">TRUE</span>                     </span>
<span id="cb51-19"><a href="#cb51-19" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb51-20"><a href="#cb51-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb51-21"><a href="#cb51-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">Depth =</span> <span class="fu">as.integer</span>(Depth),</span>
<span id="cb51-22"><a href="#cb51-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">Iter  =</span> <span class="fu">as.integer</span>(Iter)</span>
<span id="cb51-23"><a href="#cb51-23" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb51-24"><a href="#cb51-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-25"><a href="#cb51-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-26"><a href="#cb51-26" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(obfe_long, <span class="fu">aes</span>(</span>
<span id="cb51-27"><a href="#cb51-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">x    =</span> <span class="fu">factor</span>(Depth),                   </span>
<span id="cb51-28"><a href="#cb51-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">y    =</span> observed_features,</span>
<span id="cb51-29"><a href="#cb51-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> Type                         </span>
<span id="cb51-30"><a href="#cb51-30" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb51-31"><a href="#cb51-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.8</span>),</span>
<span id="cb51-32"><a href="#cb51-32" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb51-33"><a href="#cb51-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(</span>
<span id="cb51-34"><a href="#cb51-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(</span>
<span id="cb51-35"><a href="#cb51-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">x      =</span> <span class="fu">as.numeric</span>(<span class="fu">factor</span>(Depth)),</span>
<span id="cb51-36"><a href="#cb51-36" aria-hidden="true" tabindex="-1"></a>      <span class="at">colour =</span> Type,</span>
<span id="cb51-37"><a href="#cb51-37" aria-hidden="true" tabindex="-1"></a>      <span class="at">group  =</span> Type</span>
<span id="cb51-38"><a href="#cb51-38" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb51-39"><a href="#cb51-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun    =</span> median,</span>
<span id="cb51-40"><a href="#cb51-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom   =</span> <span class="st">"line"</span>,</span>
<span id="cb51-41"><a href="#cb51-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">size   =</span> <span class="dv">1</span></span>
<span id="cb51-42"><a href="#cb51-42" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb51-43"><a href="#cb51-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Sequencing Depth"</span>) <span class="sc">+</span></span>
<span id="cb51-44"><a href="#cb51-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Observed Features"</span>) <span class="sc">+</span></span>
<span id="cb51-45"><a href="#cb51-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb51-46"><a href="#cb51-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb51-47"><a href="#cb51-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb51-48"><a href="#cb51-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb51-49"><a href="#cb51-49" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb51-50"><a href="#cb51-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>( <span class="st">"#8f0052"</span>,<span class="st">"#276518"</span>)) <span class="sc">+</span></span>
<span id="cb51-51"><a href="#cb51-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>( <span class="st">"#8f0052"</span>,<span class="st">"#276518"</span>))</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-57-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="shannon-2" class="level4">
<h4 class="anchored" data-anchor-id="shannon-2">Shannon</h4>
<p>En <code>R</code>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>obfe <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">"tables/shannon.csv"</span>,</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col_types =</span> <span class="fu">cols</span>(</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>                 <span class="st">`</span><span class="at">sample-id</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">col_skip</span>(),</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">accession   =</span> <span class="fu">col_skip</span>(),</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">sra_run     =</span> <span class="fu">col_skip</span>(),</span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Type        =</span> <span class="fu">col_factor</span>(),</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">.default    =</span> <span class="fu">col_double</span>()</span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a>               ))</span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a>obfe_long <span class="ot">&lt;-</span> obfe <span class="sc">%&gt;%</span></span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"depth-"</span>), </span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="fu">c</span>(<span class="st">"Depth"</span>, <span class="st">"Iter"</span>),            </span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_pattern =</span> <span class="st">"depth-([0-9]+)_iter-([0-9]+)"</span>,</span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"Shannon_Index"</span>,</span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_drop_na =</span> <span class="cn">TRUE</span>                     </span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">Depth =</span> <span class="fu">as.integer</span>(Depth),</span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">Iter  =</span> <span class="fu">as.integer</span>(Iter)</span>
<span id="cb52-23"><a href="#cb52-23" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb52-24"><a href="#cb52-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-25"><a href="#cb52-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-26"><a href="#cb52-26" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(obfe_long, <span class="fu">aes</span>(</span>
<span id="cb52-27"><a href="#cb52-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">x    =</span> <span class="fu">factor</span>(Depth),                   </span>
<span id="cb52-28"><a href="#cb52-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">y    =</span> Shannon_Index,</span>
<span id="cb52-29"><a href="#cb52-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> Type                         </span>
<span id="cb52-30"><a href="#cb52-30" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb52-31"><a href="#cb52-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.8</span>),</span>
<span id="cb52-32"><a href="#cb52-32" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb52-33"><a href="#cb52-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(</span>
<span id="cb52-34"><a href="#cb52-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(</span>
<span id="cb52-35"><a href="#cb52-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">x      =</span> <span class="fu">as.numeric</span>(<span class="fu">factor</span>(Depth)),</span>
<span id="cb52-36"><a href="#cb52-36" aria-hidden="true" tabindex="-1"></a>      <span class="at">colour =</span> Type,</span>
<span id="cb52-37"><a href="#cb52-37" aria-hidden="true" tabindex="-1"></a>      <span class="at">group  =</span> Type</span>
<span id="cb52-38"><a href="#cb52-38" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb52-39"><a href="#cb52-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun    =</span> median,</span>
<span id="cb52-40"><a href="#cb52-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom   =</span> <span class="st">"line"</span>,</span>
<span id="cb52-41"><a href="#cb52-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">size   =</span> <span class="dv">1</span></span>
<span id="cb52-42"><a href="#cb52-42" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb52-43"><a href="#cb52-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Sequencing Depth"</span>) <span class="sc">+</span></span>
<span id="cb52-44"><a href="#cb52-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Shannon Index"</span>) <span class="sc">+</span></span>
<span id="cb52-45"><a href="#cb52-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb52-46"><a href="#cb52-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb52-47"><a href="#cb52-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb52-48"><a href="#cb52-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb52-49"><a href="#cb52-49" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb52-50"><a href="#cb52-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>( <span class="st">"#8f0052"</span>,<span class="st">"#276518"</span>)) <span class="sc">+</span></span>
<span id="cb52-51"><a href="#cb52-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>( <span class="st">"#8f0052"</span>,<span class="st">"#276518"</span>))</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-58-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="faith-pd-1" class="level3">
<h3 class="anchored" data-anchor-id="faith-pd-1">Faith-PD</h3>
<p>En <code>R</code>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>obfe <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">"tables/faith_pd.csv"</span>,</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col_types =</span> <span class="fu">cols</span>(</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>                 <span class="st">`</span><span class="at">sample-id</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">col_skip</span>(),</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">accession   =</span> <span class="fu">col_skip</span>(),</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">sra_run     =</span> <span class="fu">col_skip</span>(),</span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">Type        =</span> <span class="fu">col_factor</span>(),</span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">.default    =</span> <span class="fu">col_double</span>()</span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>               ))</span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a>obfe_long <span class="ot">&lt;-</span> obfe <span class="sc">%&gt;%</span></span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"depth-"</span>), </span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="fu">c</span>(<span class="st">"Depth"</span>, <span class="st">"Iter"</span>),            </span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_pattern =</span> <span class="st">"depth-([0-9]+)_iter-([0-9]+)"</span>,</span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"faith"</span>,</span>
<span id="cb53-18"><a href="#cb53-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_drop_na =</span> <span class="cn">TRUE</span>                     </span>
<span id="cb53-19"><a href="#cb53-19" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb53-20"><a href="#cb53-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb53-21"><a href="#cb53-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">Depth =</span> <span class="fu">as.integer</span>(Depth),</span>
<span id="cb53-22"><a href="#cb53-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">Iter  =</span> <span class="fu">as.integer</span>(Iter)</span>
<span id="cb53-23"><a href="#cb53-23" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb53-24"><a href="#cb53-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-25"><a href="#cb53-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-26"><a href="#cb53-26" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(obfe_long, <span class="fu">aes</span>(</span>
<span id="cb53-27"><a href="#cb53-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">x    =</span> <span class="fu">factor</span>(Depth),                   </span>
<span id="cb53-28"><a href="#cb53-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">y    =</span> faith,</span>
<span id="cb53-29"><a href="#cb53-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> Type                         </span>
<span id="cb53-30"><a href="#cb53-30" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb53-31"><a href="#cb53-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.8</span>),</span>
<span id="cb53-32"><a href="#cb53-32" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb53-33"><a href="#cb53-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(</span>
<span id="cb53-34"><a href="#cb53-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(</span>
<span id="cb53-35"><a href="#cb53-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">x      =</span> <span class="fu">as.numeric</span>(<span class="fu">factor</span>(Depth)),</span>
<span id="cb53-36"><a href="#cb53-36" aria-hidden="true" tabindex="-1"></a>      <span class="at">colour =</span> Type,</span>
<span id="cb53-37"><a href="#cb53-37" aria-hidden="true" tabindex="-1"></a>      <span class="at">group  =</span> Type</span>
<span id="cb53-38"><a href="#cb53-38" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb53-39"><a href="#cb53-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun    =</span> median,</span>
<span id="cb53-40"><a href="#cb53-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom   =</span> <span class="st">"line"</span>,</span>
<span id="cb53-41"><a href="#cb53-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">size   =</span> <span class="dv">1</span></span>
<span id="cb53-42"><a href="#cb53-42" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb53-43"><a href="#cb53-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Sequencing Depth"</span>) <span class="sc">+</span></span>
<span id="cb53-44"><a href="#cb53-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Faith PD"</span>) <span class="sc">+</span></span>
<span id="cb53-45"><a href="#cb53-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb53-46"><a href="#cb53-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb53-47"><a href="#cb53-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb53-48"><a href="#cb53-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb53-49"><a href="#cb53-49" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb53-50"><a href="#cb53-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>( <span class="st">"#8f0052"</span>,<span class="st">"#276518"</span>)) <span class="sc">+</span></span>
<span id="cb53-51"><a href="#cb53-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>( <span class="st">"#8f0052"</span>,<span class="st">"#276518"</span>))</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-59-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="fisher-alpha" class="level2">
<h2 class="anchored" data-anchor-id="fisher-alpha">Fisher Alpha</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha <span class="dt">\ </span>                  </span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--i-table</span> table-dada2-tomato-16s.qza <span class="dt">\ </span>              </span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--p-metric</span> fisher_alpha <span class="dt">\ </span>               </span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--o-alpha-diversity</span> fisher_alpha_diversity.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-group-significance <span class="dt">\</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>--i-alpha-diversity fisher_alpha_diversity.qza <span class="dt">\</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>--m-metadata-file Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>--o-visualization core-metrics-results/fisher_alpha-group-significance.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view core-metrics-results/fisher_alpha-group-significance.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/FISHER_ALPHA_EX1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/FISHER_ALPHA_EX2.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
<section id="section" class="level2" data-link="Entrenamiento base de datos SILVA">
<h2 data-link="Entrenamiento base de datos SILVA" class="anchored" data-anchor-id="section"></h2>
</section>
<section id="otras-métricas-de-beta-diversidad" class="level2">
<h2 class="anchored" data-anchor-id="otras-métricas-de-beta-diversidad">Otras métricas de Beta Diversidad</h2>
<p>Los siguientes datos se descargaron a partir del siguinete comando en <code>bash</code> para posteriormente gráficar en <code>R</code>.</p>
<p>En <code>bash</code>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view core-metrics-results/unweighted-unifrac-body-site-significance.qzv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="usando-r-para-gráficar" class="level3">
<h3 class="anchored" data-anchor-id="usando-r-para-gráficar">Usando R para gráficar</h3>
<section id="permanova-3" class="level4">
<h4 class="anchored" data-anchor-id="permanova-3">PERMANOVA</h4>
<p>En <code>R</code>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(<span class="st">"tables/raw_data_permanova.tsv"</span>)</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>h1 <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Group1 <span class="sc">==</span> <span class="st">"NC"</span>)</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(h1, <span class="fu">aes</span>(<span class="at">y =</span> Distance, <span class="at">x =</span> Group2, <span class="at">fill =</span> Group2)) <span class="sc">+</span></span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span> <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Distance to NC"</span>, <span class="at">fill =</span> <span class="st">"Group"</span>) <span class="sc">+</span></span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Group"</span>) <span class="sc">+</span> </span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>)) <span class="sc">+</span></span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>( <span class="st">"#8f0052"</span>,<span class="st">"#276518"</span>))</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-64-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>h2 <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Group1 <span class="sc">==</span> <span class="st">"Tomato"</span>)</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(h2, <span class="fu">aes</span>(<span class="at">y =</span> Distance, <span class="at">x =</span> Group2, <span class="at">fill =</span> Group2)) <span class="sc">+</span></span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span> <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Distance to Tomato"</span>, <span class="at">fill =</span> <span class="st">"Group"</span>) <span class="sc">+</span></span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Group"</span>) <span class="sc">+</span> </span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>)) <span class="sc">+</span></span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>( <span class="st">"#8f0052"</span>,<span class="st">"#276518"</span>))</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-64-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="arppdujupz" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#arppdujupz table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#arppdujupz thead, #arppdujupz tbody, #arppdujupz tfoot, #arppdujupz tr, #arppdujupz td, #arppdujupz th {
  border-style: none;
}

#arppdujupz p {
  margin: 0;
  padding: 0;
}

#arppdujupz .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#arppdujupz .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#arppdujupz .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#arppdujupz .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#arppdujupz .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#arppdujupz .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#arppdujupz .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#arppdujupz .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#arppdujupz .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#arppdujupz .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#arppdujupz .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#arppdujupz .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#arppdujupz .gt_spanner_row {
  border-bottom-style: hidden;
}

#arppdujupz .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#arppdujupz .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#arppdujupz .gt_from_md > :first-child {
  margin-top: 0;
}

#arppdujupz .gt_from_md > :last-child {
  margin-bottom: 0;
}

#arppdujupz .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#arppdujupz .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#arppdujupz .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#arppdujupz .gt_row_group_first td {
  border-top-width: 2px;
}

#arppdujupz .gt_row_group_first th {
  border-top-width: 2px;
}

#arppdujupz .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#arppdujupz .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#arppdujupz .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#arppdujupz .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#arppdujupz .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#arppdujupz .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#arppdujupz .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#arppdujupz .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#arppdujupz .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#arppdujupz .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#arppdujupz .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#arppdujupz .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#arppdujupz .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#arppdujupz .gt_left {
  text-align: left;
}

#arppdujupz .gt_center {
  text-align: center;
}

#arppdujupz .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#arppdujupz .gt_font_normal {
  font-weight: normal;
}

#arppdujupz .gt_font_bold {
  font-weight: bold;
}

#arppdujupz .gt_font_italic {
  font-style: italic;
}

#arppdujupz .gt_super {
  font-size: 65%;
}

#arppdujupz .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#arppdujupz .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#arppdujupz .gt_indent_1 {
  text-indent: 5px;
}

#arppdujupz .gt_indent_2 {
  text-indent: 10px;
}

#arppdujupz .gt_indent_3 {
  text-indent: 15px;
}

#arppdujupz .gt_indent_4 {
  text-indent: 20px;
}

#arppdujupz .gt_indent_5 {
  text-indent: 25px;
}

#arppdujupz .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#arppdujupz div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="Group-1" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 1</th>
<th id="Group-2" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 2</th>
<th id="Sample-size" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Sample size</th>
<th id="Permutations" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Permutations</th>
<th id="pseudo-F" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">pseudo-F</th>
<th id="p-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">p-value</th>
<th id="q-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">q-value</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Group 1">NC</td>
<td class="gt_row gt_left" headers="Group 2">Tomato</td>
<td class="gt_row gt_right" headers="Sample size">12</td>
<td class="gt_row gt_right" headers="Permutations">999</td>
<td class="gt_row gt_right" headers="pseudo-F">5.749228</td>
<td class="gt_row gt_right" headers="p-value">0.008</td>
<td class="gt_row gt_right" headers="q-value">0.008</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
</section>
</section>
<section id="entrenamiento-base-de-datos-silva" class="level2">
<h2 class="anchored" data-anchor-id="entrenamiento-base-de-datos-silva">Entrenamiento base de datos SILVA</h2>
<p><strong>NOTA</strong> ⚠️: para entrenar el clasificador debes contar con al menos <strong>8GB</strong> de RAM en tu máquina.</p>
<section id="descarga-de-la-base-de-datos-silva" class="level3">
<h3 class="anchored" data-anchor-id="descarga-de-la-base-de-datos-silva">Descarga de la base de datos SILVA</h3>
<p>Descarga la base de datos SILVA en su versión 138.2, específicamente el conjunto de datos SSURef_NR99. Este conjunto contiene secuencias del gen 16S rRNA agrupadas al 99% de similitud, lo que reduce redundancia y facilita el procesamiento.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> rescript get-silva-data <span class="dt">\</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>--p-version <span class="st">'138.2'</span> <span class="dt">\</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>--p-target <span class="st">'SSURef_NR99'</span> <span class="dt">\</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>--o-silva-sequences silva-138.2-ssu-nr99-rna-seqs.qza <span class="dt">\</span></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>--o-silva-taxonomy silva-138.2-ssu-nr99-tax.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="conversión-de-rna-a-dna" class="level3">
<h3 class="anchored" data-anchor-id="conversión-de-rna-a-dna">Conversión de RNA a DNA</h3>
<p>Convierte las secuencias de RNA descargadas a secuencias de DNA. Esto es necesario porque los análisis posteriores, incluyendo el entrenamiento del clasificador.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> rescript reverse-transcribe <span class="dt">\</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>--i-rna-sequences silva-138.2-ssu-nr99-rna-seqs.qza <span class="dt">\ </span>   <span class="at">--o-dna-sequences</span> silva-138.2-ssu-nr99-seqs.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="eliminación-de-secuencias-de-baja-calidad" class="level3">
<h3 class="anchored" data-anchor-id="eliminación-de-secuencias-de-baja-calidad">Eliminación de secuencias de baja calidad</h3>
<p>Filtra y elimina secuencias de baja calidad o con posibles errores, como aquellas con bases ambiguas (ej. ‘N’) o secuencias con regiones repetitivas largas que podrían afectar la precisión del clasificador.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> rescript cull-seqs <span class="dt">\</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>--i-sequences silva-138.2-ssu-nr99-seqs.qza <span class="dt">\</span></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>--o-clean-sequences silva-138.2-ssu-nr99-seqs-cleaned.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="filtrado-de-secuencias-por-longitud-y-taxón" class="level3">
<h3 class="anchored" data-anchor-id="filtrado-de-secuencias-por-longitud-y-taxón">Filtrado de secuencias por longitud y taxón</h3>
<p>Filtra las secuencias según su longitud mínima, dependiendo del grupo taxonómico:</p>
<ul>
<li><p>Archaea: ≥900 bp</p></li>
<li><p>Bacteria: ≥1200 bp</p></li>
<li><p>Eukaryota: ≥1400 bp</p></li>
</ul>
<p>Esto asegura que solo se utilicen secuencias completas o casi completas.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> rescript filter-seqs-length-by-taxon <span class="dt">\</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>--i-sequences silva-138.2-ssu-nr99-seqs-cleaned.qza <span class="dt">\</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>--i-taxonomy silva-138.2-ssu-nr99-tax.qza <span class="dt">\</span></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>--p-labels Archaea Bacteria Eukaryota <span class="dt">\</span></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>--p-min-lens 900 1200 1400 <span class="dt">\</span></span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>--o-filtered-seqs silva-138.2-ssu-nr99-seqs-filt.qza <span class="dt">\ </span>   <span class="at">--o-discarded-seqs</span> silva-138.2-ssu-nr99-seqs-discard.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="dereplicación-de-secuencias" class="level3">
<h3 class="anchored" data-anchor-id="dereplicación-de-secuencias">Dereplicación de secuencias</h3>
<p>Elimina secuencias duplicadas, manteniendo solo las únicas junto con su información taxonómica. Esto reduce la carga computacional sin perder diversidad taxonómica.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> rescript dereplicate <span class="dt">\</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>--i-sequences silva-138.2-ssu-nr99-seqs-filt.qza  <span class="dt">\</span></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>--i-taxa silva-138.2-ssu-nr99-tax.qza <span class="dt">\</span></span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>--p-mode <span class="st">'uniq'</span> <span class="dt">\</span></span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>--o-dereplicated-sequences silva-138.2-ssu-nr99-seqs-derep-uniq.qza <span class="dt">\</span></span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>--o-dereplicated-taxa silva-138.2-ssu-nr99-tax-derep-uniq.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="extracción-de-la-región-v4" class="level3">
<h3 class="anchored" data-anchor-id="extracción-de-la-región-v4">Extracción de la región V4</h3>
<p>Extrae la región hipervariable V4 del gen 16S rRNA usando los cebadores 515F (forward) y 806R (reverse). Esta región es comúnmente utilizada en estudios de microbiomas, por lo que el clasificador se optimiza para datos que la contengan.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier extract-reads <span class="dt">\</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>--i-sequences silva-138.2-ssu-nr99-seqs-derep-uniq.qza <span class="dt">\</span></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>--p-f-primer GTGYCAGCMGCCGCGGTAA <span class="dt">\</span></span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>--p-r-primer GGACTACNVGGGTWTCTAAT <span class="dt">\</span></span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>--p-n-jobs 5 <span class="dt">\</span></span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>--p-read-orientation <span class="st">'forward'</span> <span class="dt">\</span></span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a>--o-reads silva-138.2-ssu-nr99-seqs-515f-806r.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="dereplicación-de-la-región-v4" class="level3">
<h3 class="anchored" data-anchor-id="dereplicación-de-la-región-v4">Dereplicación de la región V4</h3>
<p>Realiza una segunda dereplicación, esta vez sobre las secuencias de la región V4, eliminando duplicados adicionales que puedan haber surgido tras la extracción.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> rescript dereplicate <span class="dt">\</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>--i-sequences silva-138.2-ssu-nr99-seqs-515f-806r.qza <span class="dt">\</span></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>--i-taxa silva-138.2-ssu-nr99-tax-derep-uniq.qza <span class="dt">\</span></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>--p-mode <span class="st">'uniq'</span> <span class="dt">\</span></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>--o-dereplicated-sequences silva-138.2-ssu-nr99-seqs-515f-806r-uniq.qza <span class="dt">\</span></span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>--o-dereplicated-taxa  silva-138.2-ssu-nr99-tax-515f-806r-derep-uniq.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="entrenamiento-del-clasificador-naive-bayes" class="level3">
<h3 class="anchored" data-anchor-id="entrenamiento-del-clasificador-naive-bayes">Entrenamiento del clasificador Naive Bayes</h3>
<p>Entrena un clasificador Naive Bayes utilizando las secuencias únicas de la región V4 y su taxonomía. Este clasificador puede usarse luego para asignar taxonomía a secuencias 16S de muestras de microbiomas.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier fit-classifier-naive-bayes <span class="dt">\</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>--i-reference-reads silva-138.2-ssu-nr99-seqs-515f-806r-uniq.qza <span class="dt">\</span></span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>--i-reference-taxonomy silva-138.2-ssu-nr99-tax-515f-806r-derep-uniq.qza <span class="dt">\</span></span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>--o-classifier silva-138.2-ssu-nr99-515f-806r-classifier.qza</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="boxplots-para-algunos-clados-importantes" class="level2">
<h2 class="anchored" data-anchor-id="boxplots-para-algunos-clados-importantes">Boxplots para algunos clados importantes</h2>
<p>Para este punto bajamos los archivos .csv del nivel 6 al acceder al archivo <code>taxa-bar-plots.qzv</code></p>
<p>Para este ejemplo vamos a explorar los Generos <em>Lysobacteraceae</em> y <em>Micropepsis.</em></p>
<section id="lysobacteraceae" class="level3">
<h3 class="anchored" data-anchor-id="lysobacteraceae"><strong><em>Lysobacteraceae</em></strong></h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">"tables/level-6.csv"</span>)</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>lysobacter <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[stringr<span class="sc">::</span><span class="fu">str_detect</span>(<span class="fu">colnames</span>(df), <span class="st">"Lysobacteraceae"</span>)]</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>sub_df <span class="ot">&lt;-</span> df[, <span class="fu">c</span>(<span class="st">"Type"</span>, lysobacter)]</span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sub_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Type"</span>, <span class="st">"Lyso1"</span>, <span class="st">"Lyso2"</span>, <span class="st">"Lyso3"</span>)</span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a>Lyso_df <span class="ot">&lt;-</span> sub_df <span class="sc">%&gt;%</span> <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Lysobacteraceae =</span> <span class="fu">sum</span>(<span class="fu">c</span>(Lyso1, Lyso2, Lyso3)))</span>
<span id="cb68-12"><a href="#cb68-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-13"><a href="#cb68-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb68-14"><a href="#cb68-14" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggboxplot</span>(Lyso_df, <span class="at">x =</span> <span class="st">"Type"</span>, <span class="at">y =</span> <span class="st">"Lysobacteraceae"</span>,</span>
<span id="cb68-15"><a href="#cb68-15" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">palette =</span><span class="fu">c</span>(<span class="st">"#8f0052"</span>,<span class="st">"#276518"</span>),</span>
<span id="cb68-16"><a href="#cb68-16" aria-hidden="true" tabindex="-1"></a>               <span class="at">add =</span> <span class="st">"jitter"</span>, <span class="at">shape =</span> <span class="st">"Type"</span>,</span>
<span id="cb68-17"><a href="#cb68-17" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">"Type"</span>)</span>
<span id="cb68-18"><a href="#cb68-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-19"><a href="#cb68-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-20"><a href="#cb68-20" aria-hidden="true" tabindex="-1"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="fu">c</span>(<span class="st">"NC"</span>, <span class="st">"Tomato"</span>))</span>
<span id="cb68-21"><a href="#cb68-21" aria-hidden="true" tabindex="-1"></a>p <span class="sc">+</span> <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> my_comparisons, <span class="at">label =</span> <span class="st">"p.signif"</span>, <span class="at">paired =</span> <span class="cn">TRUE</span>, <span class="at">method =</span> <span class="st">"t.test"</span>)<span class="sc">+</span> </span>
<span id="cb68-22"><a href="#cb68-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb68-23"><a href="#cb68-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>)) <span class="sc">+</span></span>
<span id="cb68-24"><a href="#cb68-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Lysobacteraceae ASV"</span>)</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-74-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="micropepsis" class="level3">
<h3 class="anchored" data-anchor-id="micropepsis"><strong><em>Micropepsis</em></strong></h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">"tables/level-6.csv"</span>)</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>Micropepsis <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[stringr<span class="sc">::</span><span class="fu">str_detect</span>(<span class="fu">colnames</span>(df), <span class="st">"Micropepsis"</span>)]</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>sub_df <span class="ot">&lt;-</span> df[, <span class="fu">c</span>(<span class="st">"Type"</span>, Micropepsis)]</span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sub_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Type"</span>, <span class="st">"micropepsis"</span>)</span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggboxplot</span>(sub_df, <span class="at">x =</span> <span class="st">"Type"</span>, <span class="at">y =</span> <span class="st">"micropepsis"</span>,</span>
<span id="cb69-12"><a href="#cb69-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">palette =</span><span class="fu">c</span>(<span class="st">"#8f0052"</span>,<span class="st">"#276518"</span>),</span>
<span id="cb69-13"><a href="#cb69-13" aria-hidden="true" tabindex="-1"></a>               <span class="at">add =</span> <span class="st">"jitter"</span>, <span class="at">shape =</span> <span class="st">"Type"</span>,</span>
<span id="cb69-14"><a href="#cb69-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">"Type"</span>)</span>
<span id="cb69-15"><a href="#cb69-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-16"><a href="#cb69-16" aria-hidden="true" tabindex="-1"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="fu">c</span>(<span class="st">"NC"</span>, <span class="st">"Tomato"</span>))</span>
<span id="cb69-17"><a href="#cb69-17" aria-hidden="true" tabindex="-1"></a>p <span class="sc">+</span> <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> my_comparisons, <span class="at">label =</span> <span class="st">"p.signif"</span>, <span class="at">paired =</span> <span class="cn">TRUE</span>, </span>
<span id="cb69-18"><a href="#cb69-18" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">"t.test"</span>)<span class="sc">+</span> </span>
<span id="cb69-19"><a href="#cb69-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb69-20"><a href="#cb69-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>)) <span class="sc">+</span></span>
<span id="cb69-21"><a href="#cb69-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Micropepsis ASV"</span>)</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-75-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copiado");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copiado");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




<script src="index_files/libs/quarto-html/zenscroll-min.js"></script>
</body></html>