<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="OMICs analysis">

<title>Metabarcoding: Tomate 16S usando Qiime2</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="tomato_16S_files/libs/clipboard/clipboard.min.js"></script>
<script src="tomato_16S_files/libs/quarto-html/quarto.js"></script>
<script src="tomato_16S_files/libs/quarto-html/popper.min.js"></script>
<script src="tomato_16S_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="tomato_16S_files/libs/quarto-html/anchor.min.js"></script>
<link href="tomato_16S_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="tomato_16S_files/libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="tomato_16S_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="tomato_16S_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="tomato_16S_files/libs/bootstrap/bootstrap-a71a25f19e128005acb9c00d38d9c2cf.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Metabarcoding: Tomate 16S usando Qiime2</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>OMICs analysis </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="tomate-16s" class="level2">
<h2 class="anchored" data-anchor-id="tomate-16s">Tomate 16S</h2>
<p>Datos de partida “<a href="https://www.ncbi.nlm.nih.gov/sra/?term=SRX13358039">Disentangling the genetic basis of rhizosphere microbiome assembly in tomato</a>”</p>
<p>Para este tutorial vamos a utilizar <a href="https://qiime2.org/">Qiime2</a><strong>,</strong> <a href="https://library.qiime2.org/quickstart">aquí</a> lo puedes instalar según sea tu sistema operativo. Para los siguientes pasos vamos a autilizar la distribución “<a href="https://amplicon-docs.qiime2.org/">amplicon</a>”. Este tutorial se hizo utilizando el sistema operativo <code>Ubuntu 22.04.5 LTS</code>, en un equipo con <code>24 Gb</code> de <code>RAM</code> <code>Intel® Core™ i7-3610QM CPU @ 2.30GHz × 8</code>. Todo se hizo usando la <code>terminal</code>.</p>
<p>Una vez instalado <code>qiime</code> activa el ambiente conda.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate qiime2-amplicon-2024.10</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Primer paso:</strong> Genera una carpeta (directorio) exclusiva para llevar a cabo el proyecto, en este caso el nombre de la carpeta es <code>2025_Demo_tomato_16S</code>, en ella se va generar una carpeta para guardar los genomas 16S que tendrá por nombre <code>data</code>.</p>
<p>Lo genomas 16S los podrás encontrar <a href="https://drive.google.com/drive/folders/1mQRMfCqHzCaF9_sc_az1Cj3l6YCAAB5V?usp=drive_link"><strong>aquí</strong></a>para descargar.</p>
<p>A continuación.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> 2025_Demo_tomato_16S</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> 2025_Demo_tomato_16S</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> data</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="importar-datos-pareados-con-qiime2" class="level2">
<h2 class="anchored" data-anchor-id="importar-datos-pareados-con-qiime2">Importar datos pareados con Qiime2</h2>
<p>Vamos a generar un archivo <code>manifiesto</code> en el vamos a indicar la localización de las lecturas pareadas. El archivo se ve de la siguiente manera:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 12 × 3
   `sample-id` `forward-absolute-filepath`    `reverse-absolute-filepath`   
   &lt;chr&gt;       &lt;chr&gt;                          &lt;chr&gt;                         
 1 Tomato_1    $PWD/data/Tomato_1_R1.fastq.gz $PWD/data/Tomato_1_R2.fastq.gz
 2 Tomato_2    $PWD/data/Tomato_2_R1.fastq.gz $PWD/data/Tomato_2_R2.fastq.gz
 3 Tomato_3    $PWD/data/Tomato_3_R1.fastq.gz $PWD/data/Tomato_3_R2.fastq.gz
 4 Tomato_4    $PWD/data/Tomato_4_R1.fastq.gz $PWD/data/Tomato_4_R2.fastq.gz
 5 Tomato_5    $PWD/data/Tomato_5_R1.fastq.gz $PWD/data/Tomato_5_R2.fastq.gz
 6 Tomato_6    $PWD/data/Tomato_6_R1.fastq.gz $PWD/data/Tomato_6_R2.fastq.gz
 7 NC_1        $PWD/data/NC_1_R1.fastq.gz     $PWD/data/NC_1_R2.fastq.gz    
 8 NC_2        $PWD/data/NC_2_R1.fastq.gz     $PWD/data/NC_2_R2.fastq.gz    
 9 NC_3        $PWD/data/NC_3_R1.fastq.gz     $PWD/data/NC_3_R2.fastq.gz    
10 NC_4        $PWD/data/NC_4_R1.fastq.gz     $PWD/data/NC_4_R2.fastq.gz    
11 NC_5        $PWD/data/NC_5_R1.fastq.gz     $PWD/data/NC_5_R2.fastq.gz    
12 NC_6        $PWD/data/NC_6_R1.fastq.gz     $PWD/data/NC_6_R2.fastq.gz    </code></pre>
</div>
</div>
<p>Este archivo lo vamos a nombrar como <code>tomato_manifest</code>, este archivo es de tipo <code>tab-separate-value</code>. Puedes descargar el archivo <code>manifiesto</code> <a href="https://drive.google.com/file/d/1hIzzApEfZfKqOlxhjVDsSvKVmqUbEA1d/view?usp=drive_link">aquí</a>.</p>
<p>Para importar los secuencias 16S pareadas vamos a utilizar el siguiente código:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools import <span class="dt">\</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>--type <span class="st">'SampleData[PairedEndSequencesWithQuality]'</span> <span class="dt">\</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>--input-path tomato_manifest <span class="dt">\</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>--output-path paired-end-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>--input-format PairedEndFastqManifestPhred33V2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Una vez importadas las secuencias, vamos a realizar un preprocesmaiento con <a href="https://cutadapt.readthedocs.io/en/stable/"><code>cutadapt</code></a> para eliminar adapdores, <code>qiime2</code> nos permite hacer esto utilizando la siguiente instrucción:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> cutadapt trim-paired <span class="dt">\</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>--i-demultiplexed-sequences paired-end-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>--p-front-f CCTACGGGNGGCWGCAG <span class="dt">\</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>--p-adapter-f GGATTAGATACCCBDGTAGTC <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>--p-front-r GACTACHVGGGTATCTAATCC <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>--p-adapter-r CTGCWGCCNCCCGTAGG <span class="dt">\</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>--p-match-read-wildcards <span class="dt">\</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>--p-discard-untrimmed <span class="dt">\</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>--o-trimmed-sequences trimmed_paired-end-tomato-16s.qza <span class="at">--p-cores</span> 5</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="explicación-del-comando" class="level3">
<h3 class="anchored" data-anchor-id="explicación-del-comando">Explicación del comando:</h3>
<ul>
<li><p><strong><code>--i-demultiplexed-sequences paired-end-tomato-16s.qza</code></strong><br>
</p>
<p>Este parámetro indica el archivo de entrada que contiene tus secuencias pareadas ya demultiplexadas. El archivo de entrada es <code>paired-end-tomato-16s.qza</code>.</p></li>
<li><p><strong>Adaptadores para las lecturas forward y reverse</strong></p>
<ul>
<li><p><strong><code>--p-front-f CCTACGGGNGGCWGCAG</code></strong>: Adaptador 5’ de las lecturas forward.</p></li>
<li><p><strong><code>--p-adapter-f GGATTAGATACCCBDGTAGTC</code></strong>: Adaptador 3’ de las lecturas forward.</p></li>
<li><p><strong><code>--p-front-r GACTACHVGGGTATCTAATCC</code></strong>: Adaptador 5’ de las lecturas reverse.</p></li>
<li><p><strong><code>--p-adapter-r CTGCWGCCNCCCGTAGG</code></strong>: Adaptador 3’ de las lecturas reverse.</p></li>
</ul></li>
<li><p><strong><code>--p-match-read-wildcards</code></strong><br>
</p>
<p>Este flag permite que los comodines (como N, W, B, D, H, V) en los adaptadores coincidan con las bases en las secuencias de lectura.</p></li>
<li><p><strong><code>--p-discard-untrimmed</code></strong><br>
</p>
<p>Esta opción descarta las secuencias que no contienen los adaptadores especificados, asegurando que solo se conserven las lecturas correctamente recortadas.</p></li>
<li><p><strong><code>--o-trimmed-sequences trimmed.qza</code></strong><br>
</p>
<p>Define el archivo de salida en formato <code>QZA</code> (<code>trimmed_paired-end-tomato-16s.qza</code>) que contendrá las secuencias recortadas. <code>Qiime</code> genera un solo archivo <code>QZA</code> que puedes usar en pasos posteriores del análisis.</p></li>
</ul>
</section>
</section>
<section id="explorar-los-archivos-importados" class="level2">
<h2 class="anchored" data-anchor-id="explorar-los-archivos-importados">Explorar los archivos importados</h2>
<p>Qiime permite generar un reporte HTML para ver la cantidad de lecturas por muestras y métricas generales de secuencia. Para ello podemos generar el siguiente archivo:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> demux summarize <span class="dt">\</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>--i-data trimmed_paired-end-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>--o-visualization summary_tomato_16s.qzv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Para visualizar este archivo utiliza el siguiente código:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view summary_tomato_16s.qzv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="dada2" class="level2">
<h2 class="anchored" data-anchor-id="dada2"><a href="https://benjjneb.github.io/dada2/">DADA2</a></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> dada2 denoise-paired <span class="dt">\</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>--i-demultiplexed-seqs trimmed_paired-end-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>--p-trunc-len-f 280 <span class="dt">\</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>--p-trunc-len-r 260 <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>--o-representative-sequences rep-seqs-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>--o-table table-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>--o-denoising-stats stats-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>--p-n-threads 5</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Con esto podemos ver un reporte de la cantidad de lecturas filtradas:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> metadata tabulate <span class="dt">\</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--m-input-file</span> stats-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> stats-dada2-tomato-16s.qzv</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools view stats-dada2-tomato-16s.qzv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Después de terminar el paso de filtrado de calidad, querrás explorar los datos que obtuviste. Este filtrado es importante porque elimina las secuencias de baja calidad o con errores, dejándote datos más confiables para trabajar. Para revisar esos datos, puedes usar dos comandos que te muestran resúmenes visuales fáciles de entender.</p>
<p>EXPERIMENT DESIGN</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 13 × 4
   `Sample-id` accession   sra_run     Type       
   &lt;chr&gt;       &lt;chr&gt;       &lt;chr&gt;       &lt;chr&gt;      
 1 #q2:types   categorical categorical categorical
 2 Tomato_1    SRX13358040 SRR17174837 Tomato     
 3 Tomato_2    SRX13358039 SRR17174838 Tomato     
 4 Tomato_3    SRX13358038 SRR17174839 Tomato     
 5 Tomato_4    SRX13358037 SRR17174840 Tomato     
 6 Tomato_5    SRX13358036 SRR17174841 Tomato     
 7 Tomato_6    SRX13358035 SRR17174842 Tomato     
 8 NC_1        SRX13358032 SRR17174845 NC         
 9 NC_2        SRX13358026 SRR17174851 NC         
10 NC_3        SRX13358024 SRR17174853 NC         
11 NC_4        SRX13358022 SRR17174855 NC         
12 NC_5        SRX13358630 SRR17174856 NC         
13 NC_6        SRX13358517 SRR17174969 NC         </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-table summarize <span class="dt">\</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-table</span> table-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> table-dada2-tomato-16s.qzv <span class="dt">\</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--m-sample-metadata-file</span> Experiment_Design.tsv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-table tabulate-seqs <span class="dt">\</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-data</span> rep-seqs-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> rep-seqs-dada2-tomato-16s.qzv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li><p><strong>El comando <code>feature-table summarize</code></strong>, para cuando se cuenta con <code>metadata.</code><br>
</p>
<p>Este comando te ayuda a ver:</p>
<ul>
<li><p><strong>Cuántas secuencias hay en cada muestra</strong>: por ejemplo, cuántas secuencias tiene cada una de tus muestras biológicas.</p></li>
<li><p><strong>Cuántas secuencias hay por característica</strong>: las características pueden ser cosas como tipos de bacterias o genes que encontraste, en caso de agregar metadatos.</p></li>
<li><p><strong>Gráficos (histogramas)</strong>: estos te muestran cómo se distribuyen las secuencias.</p></li>
<li><p><strong>Estadísticas básicas</strong>: algunos números que resumen tus datos de forma general, para que tengas una idea rápida de lo que tienes.</p></li>
</ul></li>
<li><p><strong>El comando <code>feature-table tabulate-seqs</code></strong><br>
</p>
<p>Este comando es útil porque:</p>
<ul>
<li><p>Te da una lista que conecta cada <strong>ID de característica</strong> (un código único para cada característica) con la secuencia de ADN que representa.</p></li>
<li><p>Además, te ofrece <strong>enlaces</strong> para buscar esas secuencias en una base de datos grande llamada <strong>NCBI nt</strong>.</p></li>
</ul></li>
</ul>
<section id="por-qué-son-útiles-estos-comandos" class="level3">
<h3 class="anchored" data-anchor-id="por-qué-son-útiles-estos-comandos">¿Por qué son útiles estos comandos?</h3>
<ul>
<li><p>El primero te da una visión general de tus datos: cuántas secuencias tienes y cómo están organizadas.</p></li>
<li><p>El segundo te permite investigar a fondo las características que te interesen, conectando los códigos con las secuencias reales y dándote herramientas para explorar más.</p></li>
</ul>
</section>
</section>
<section id="medidas-de-diversidad" class="level1">
<h1>Medidas de diversidad</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity core-metrics-phylogenetic <span class="dt">\</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>--i-phylogeny rooted-tree-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>--i-table table-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>--p-sampling-depth 16890 <span class="dt">\</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>--m-metadata-file Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>--output-dir core-metrics-results</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="alfa-diversidad-α-diversidad" class="level2">
<h2 class="anchored" data-anchor-id="alfa-diversidad-α-diversidad">Alfa diversidad (α-diversidad)</h2>
<p>La <strong>alfa diversidad</strong> es la diversidad media de especies microbianas en un lugar a escala local. Representa la riqueza de especies y su abundancia relativa dentro de una sola muestra o hábitat. Es uno de los componentes fundamentales de la diversidad biológica junto con la diversidad beta y gamma.</p>
<section id="características-principales" class="level4">
<h4 class="anchored" data-anchor-id="características-principales">Características principales:</h4>
<ul>
<li>Mide la diversidad dentro de una muestra individual</li>
<li>Considera tanto el número de especies como su abundancia relativa</li>
<li>Es independiente de la escala espacial específica</li>
<li>Fundamental para caracterizar comunidades microbianas</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-group-significance <span class="dt">\</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>--i-alpha-diversity core-metrics-results/faith_pd_vector.qza <span class="dt">\</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>--m-metadata-file Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>--o-visualization core-metrics-results/faith-pd-group-significance.qzv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-group-significance <span class="dt">\</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>--i-alpha-diversity core-metrics-results/evenness_vector.qza <span class="dt">\</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>--m-metadata-file Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>--o-visualization core-metrics-results/evenness-group-significance.qzv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="eveness-equitabilidad" class="level3">
<h3 class="anchored" data-anchor-id="eveness-equitabilidad">Eveness (equitabilidad)</h3>
<p>La <strong>Evenness</strong> o <strong>equitabilidad</strong> mide qué tan uniformemente están distribuidas las abundancias de las especies en una comunidad. Una alta equitabilidad indica que las especies tienen abundancias similares.</p>
<section id="índice-de-equitabilidad-de-pielou" class="level4">
<h4 class="anchored" data-anchor-id="índice-de-equitabilidad-de-pielou">Índice de Equitabilidad de Pielou:</h4>
<p><span class="math display">\[J' = \frac{H'}{H'_{\max}} = \frac{H'}{\ln S}\]</span></p>
<p>donde: - <span class="math inline">\(J'\)</span> = índice de equitabilidad de Pielou - <span class="math inline">\(H'\)</span> = índice de Shannon observado - <span class="math inline">\(H'_{\max}\)</span> = diversidad máxima posible = <span class="math inline">\(\ln S\)</span> - <span class="math inline">\(S\)</span> = número de especies</p>
</section>
<section id="interpretación" class="level4">
<h4 class="anchored" data-anchor-id="interpretación">Interpretación:</h4>
<ul>
<li>Valores entre 0 y 1</li>
<li>0 = máxima inequidad (una especie domina completamente)</li>
<li>1 = máxima equidad (todas las especies igualmente abundantes)</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/alpha-compare.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="gqyqapcypa" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#gqyqapcypa table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#gqyqapcypa thead, #gqyqapcypa tbody, #gqyqapcypa tfoot, #gqyqapcypa tr, #gqyqapcypa td, #gqyqapcypa th {
  border-style: none;
}

#gqyqapcypa p {
  margin: 0;
  padding: 0;
}

#gqyqapcypa .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#gqyqapcypa .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#gqyqapcypa .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#gqyqapcypa .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#gqyqapcypa .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gqyqapcypa .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gqyqapcypa .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gqyqapcypa .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#gqyqapcypa .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#gqyqapcypa .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#gqyqapcypa .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#gqyqapcypa .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#gqyqapcypa .gt_spanner_row {
  border-bottom-style: hidden;
}

#gqyqapcypa .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#gqyqapcypa .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#gqyqapcypa .gt_from_md > :first-child {
  margin-top: 0;
}

#gqyqapcypa .gt_from_md > :last-child {
  margin-bottom: 0;
}

#gqyqapcypa .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#gqyqapcypa .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#gqyqapcypa .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#gqyqapcypa .gt_row_group_first td {
  border-top-width: 2px;
}

#gqyqapcypa .gt_row_group_first th {
  border-top-width: 2px;
}

#gqyqapcypa .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gqyqapcypa .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#gqyqapcypa .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#gqyqapcypa .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gqyqapcypa .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gqyqapcypa .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#gqyqapcypa .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#gqyqapcypa .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#gqyqapcypa .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gqyqapcypa .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gqyqapcypa .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#gqyqapcypa .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gqyqapcypa .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#gqyqapcypa .gt_left {
  text-align: left;
}

#gqyqapcypa .gt_center {
  text-align: center;
}

#gqyqapcypa .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#gqyqapcypa .gt_font_normal {
  font-weight: normal;
}

#gqyqapcypa .gt_font_bold {
  font-weight: bold;
}

#gqyqapcypa .gt_font_italic {
  font-style: italic;
}

#gqyqapcypa .gt_super {
  font-size: 65%;
}

#gqyqapcypa .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#gqyqapcypa .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#gqyqapcypa .gt_indent_1 {
  text-indent: 5px;
}

#gqyqapcypa .gt_indent_2 {
  text-indent: 10px;
}

#gqyqapcypa .gt_indent_3 {
  text-indent: 15px;
}

#gqyqapcypa .gt_indent_4 {
  text-indent: 20px;
}

#gqyqapcypa .gt_indent_5 {
  text-indent: 25px;
}

#gqyqapcypa .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#gqyqapcypa div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="Group-1" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 1</th>
<th id="Group-2" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 2</th>
<th id="H" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">H</th>
<th id="p-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">p-value</th>
<th id="q-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">q-value</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Group 1">NC (n=6)</td>
<td class="gt_row gt_left" headers="Group 2">Tomato (n=6)</td>
<td class="gt_row gt_right" headers="H">3.692308</td>
<td class="gt_row gt_right" headers="p-value">0.05466394</td>
<td class="gt_row gt_right" headers="q-value">0.05466394</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
</section>
<section id="faiths-phylogenetic-diversity-pd" class="level3">
<h3 class="anchored" data-anchor-id="faiths-phylogenetic-diversity-pd">Faith’s Phylogenetic Diversity (PD)</h3>
<p><strong>Faith’s Phylogenetic Diversity (PD)</strong> es un índice que incorpora información filogenética para medir la diversidad evolutiva de una comunidad. No solo considera el número de especies sino también sus relaciones evolutivas.</p>
<section id="fórmula-de-faith-pd" class="level4">
<h4 class="anchored" data-anchor-id="fórmula-de-faith-pd">Fórmula de Faith PD:</h4>
<p><span class="math display">\[PD = \sum_{i=1}^{N} L_i\]</span></p>
<p>donde: - <span class="math inline">\(PD\)</span> = diversidad filogenética de Faith - <span class="math inline">\(N\)</span> = número de ramas del árbol filogenético que conectan las especies presentes - <span class="math inline">\(L_i\)</span> = longitud de la rama <span class="math inline">\(i\)</span> en el árbol filogenético - Las ramas se suman desde las especies hasta la raíz del árbol</p>
</section>
<section id="características-principales-1" class="level4">
<h4 class="anchored" data-anchor-id="características-principales-1">Características principales:</h4>
<ul>
<li>Incorpora historia evolutiva</li>
<li>Útil para prioridades de conservación</li>
<li>Medida en unidades de tiempo evolutivo</li>
<li>Mayor PD indica mayor diversidad evolutiva</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/alpha-compare_faith.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="bklogibioa" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#bklogibioa table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#bklogibioa thead, #bklogibioa tbody, #bklogibioa tfoot, #bklogibioa tr, #bklogibioa td, #bklogibioa th {
  border-style: none;
}

#bklogibioa p {
  margin: 0;
  padding: 0;
}

#bklogibioa .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#bklogibioa .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#bklogibioa .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#bklogibioa .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#bklogibioa .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#bklogibioa .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bklogibioa .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#bklogibioa .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#bklogibioa .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#bklogibioa .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#bklogibioa .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#bklogibioa .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#bklogibioa .gt_spanner_row {
  border-bottom-style: hidden;
}

#bklogibioa .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#bklogibioa .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#bklogibioa .gt_from_md > :first-child {
  margin-top: 0;
}

#bklogibioa .gt_from_md > :last-child {
  margin-bottom: 0;
}

#bklogibioa .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#bklogibioa .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#bklogibioa .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#bklogibioa .gt_row_group_first td {
  border-top-width: 2px;
}

#bklogibioa .gt_row_group_first th {
  border-top-width: 2px;
}

#bklogibioa .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#bklogibioa .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#bklogibioa .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#bklogibioa .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bklogibioa .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#bklogibioa .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#bklogibioa .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#bklogibioa .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#bklogibioa .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bklogibioa .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#bklogibioa .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#bklogibioa .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#bklogibioa .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#bklogibioa .gt_left {
  text-align: left;
}

#bklogibioa .gt_center {
  text-align: center;
}

#bklogibioa .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#bklogibioa .gt_font_normal {
  font-weight: normal;
}

#bklogibioa .gt_font_bold {
  font-weight: bold;
}

#bklogibioa .gt_font_italic {
  font-style: italic;
}

#bklogibioa .gt_super {
  font-size: 65%;
}

#bklogibioa .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#bklogibioa .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#bklogibioa .gt_indent_1 {
  text-indent: 5px;
}

#bklogibioa .gt_indent_2 {
  text-indent: 10px;
}

#bklogibioa .gt_indent_3 {
  text-indent: 15px;
}

#bklogibioa .gt_indent_4 {
  text-indent: 20px;
}

#bklogibioa .gt_indent_5 {
  text-indent: 25px;
}

#bklogibioa .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#bklogibioa div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="Group-1" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 1</th>
<th id="Group-2" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 2</th>
<th id="H" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">H</th>
<th id="p-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">p-value</th>
<th id="q-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">q-value</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Group 1">NC (n=6)</td>
<td class="gt_row gt_left" headers="Group 2">Tomato (n=6)</td>
<td class="gt_row gt_right" headers="H">8.307692</td>
<td class="gt_row gt_right" headers="p-value">0.003947752</td>
<td class="gt_row gt_right" headers="q-value">0.003947752</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
</section>
</section>
<section id="alfa-rarefacción" class="level2">
<h2 class="anchored" data-anchor-id="alfa-rarefacción">Alfa rarefacción</h2>
<p>La <strong>alfa-rarefacción</strong> es una técnica para evaluar cómo crece la diversidad “intra-muestra” (α-diversidad) a medida que aumentas el número de lecturas secuenciadas. En metagenómica o secuenciación 16S, su objetivo principal es:</p>
<ol type="1">
<li><p><strong>Submuestrear</strong> tu biblioteca de lecturas a distintos “depths” (profundidades de secuenciación): por ejemplo, tomar aleatoriamente 100, 200, …, 4 000 lecturas de cada muestra.</p></li>
<li><p><strong>Calcular un índice de α-diversidad</strong> (observed OTUs, Shannon, Simpson, Faith’s PD, etc.) en cada submuestra.</p></li>
<li><p><strong>Trazar</strong> ese índice frente al número de lecturas: obtienes una curva de rarefacción.</p></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-rarefaction <span class="dt">\</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>--i-table table-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>--i-phylogeny rooted-tree-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>--p-max-depth 4000 <span class="dt">\</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>--m-metadata-file Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>--o-visualization alpha-rarefaction-dada2-tomato-16s.qzv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="observed-features" class="level3">
<h3 class="anchored" data-anchor-id="observed-features">Observed Features</h3>
<p>Las <strong>Observed Features</strong> representan simplemente el <strong>número de características taxonómicas diferentes</strong> (especies, géneros, familias, etc.) observadas en una muestra. Es la medida más básica de riqueza de especies.</p>
<section id="características-principales-2" class="level4">
<h4 class="anchored" data-anchor-id="características-principales-2">Características principales:</h4>
<ul>
<li>Medida directa de riqueza taxonómica</li>
<li>No considera abundancia relativa</li>
<li>Sensible al esfuerzo de muestreo</li>
<li>Base para otros índices de diversidad</li>
</ul>
<p>La fórmula es simplemente:</p>
<p><span class="math display">\[S = \text{número total de especies observadas}\]</span></p>
<p>donde: - <span class="math inline">\(S\)</span> = riqueza de especies observadas</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="tomato_16S_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="shannon" class="level3">
<h3 class="anchored" data-anchor-id="shannon">Shannon</h3>
<p>El <strong>índice de Shannon</strong> (Shannon-Weaver o Shannon-Wiener) mide la biodiversidad específica considerando tanto la riqueza de especies como su abundancia relativa. Está basado en la teoría de la información y mide la incertidumbre para predecir la especie de un individuo tomado al azar.</p>
<section id="fórmula-del-índice-de-shannon" class="level4">
<h4 class="anchored" data-anchor-id="fórmula-del-índice-de-shannon">Fórmula del índice de Shannon:</h4>
<p><span class="math display">\[H' = -\sum_{i=1}^{S} p_i \ln p_i\]</span></p>
<p>donde: - <span class="math inline">\(H'\)</span> = índice de Shannon - <span class="math inline">\(S\)</span> = número de especies (riqueza de especies) - <span class="math inline">\(p_i\)</span> = proporción de individuos de la especie <span class="math inline">\(i\)</span> respecto al total: <span class="math inline">\(p_i = \frac{n_i}{N}\)</span> - <span class="math inline">\(n_i\)</span> = número de individuos de la especie <span class="math inline">\(i\)</span> - <span class="math inline">\(N\)</span> = número total de individuos de todas las especies - <span class="math inline">\(\ln\)</span> = logaritmo natural</p>
</section>
<section id="interpretación-1" class="level4">
<h4 class="anchored" data-anchor-id="interpretación-1">Interpretación:</h4>
<ul>
<li>Valores entre 0.5 y 5 en ecosistemas naturales</li>
<li>Valores &lt; 2: baja diversidad</li>
<li>Valores &gt; 3: alta diversidad</li>
<li>Valores más altos indican mayor diversidad y equitabilidad</li>
</ul>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="tomato_16S_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="faith-pd" class="level3">
<h3 class="anchored" data-anchor-id="faith-pd">Faith PD</h3>
<p>La diversidad filogenética de Faith (PD de Faith) es una medida de la biodiversidad que cuantifica la amplitud de la historia evolutiva representada por un conjunto de especies u otras unidades taxonómicas, definida como la suma de las longitudes de las ramas de un árbol filogenético que conectan los miembros del conjunto.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="tomato_16S_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="section" class="level1">
<h1></h1>
<section id="beta-diversidad-β-diversidad" class="level2">
<h2 class="anchored" data-anchor-id="beta-diversidad-β-diversidad">Beta diversidad (β-diversidad)</h2>
<p>La <strong>beta diversidad</strong> es la relación entre la diversidad regional y la diversidad local de especies. Mide el grado de diferenciación o recambio de especies entre diferentes sitios o comunidades. En microbiología, nos permite comparar qué tan similares o diferentes son las comunidades microbianas entre distintas muestras.</p>
<section id="características-principales-3" class="level4">
<h4 class="anchored" data-anchor-id="características-principales-3">Características principales:</h4>
<ul>
<li>Mide la variación en composición entre comunidades</li>
<li>Puede deberse a recambio de especies o anidamiento</li>
<li>Valores altos indican comunidades muy diferentes</li>
<li>Valores bajos indican comunidades similares</li>
</ul>
<p>La <strong>diversidad β verdadera</strong> se calcula como:</p>
<p><span class="math display">\[\beta = \frac{\gamma}{\alpha}\]</span></p>
<p>donde: - <span class="math inline">\(\beta\)</span> = diversidad beta verdadera (número efectivo de comunidades) - <span class="math inline">\(\gamma\)</span> = diversidad total del conjunto de datos - <span class="math inline">\(\alpha\)</span> = diversidad alfa promedio</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity beta-group-significance <span class="dt">\</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>--i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza <span class="dt">\</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>--m-metadata-file Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>--m-metadata-column Type <span class="dt">\</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>--o-visualization core-metrics-results/unweighted-unifrac-body-site-significance.qzv <span class="dt">\</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>--p-pairwise</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> diversity beta-group-significance <span class="dt">\</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>--i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza <span class="dt">\</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>--m-metadata-file Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>--o-visualization core-metrics-results/unweighted-unifrac-significance.qzv <span class="dt">\</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>--p-pairwise</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="distancia-bray-curtis" class="level3">
<h3 class="anchored" data-anchor-id="distancia-bray-curtis">Distancia Bray Curtis</h3>
<p>La <strong>distancia de Bray-Curtis</strong> es una medida de disimilitud composicional entre dos sitios basada en las abundancias de especies. Es ampliamente utilizada en ecología microbiana para comparar comunidades.</p>
<section id="fórmula-de-bray-curtis" class="level4">
<h4 class="anchored" data-anchor-id="fórmula-de-bray-curtis">Fórmula de Bray-Curtis:</h4>
<p><span class="math display">\[BC_{ij} = 1 - \frac{2C_{ij}}{S_i + S_j}\]</span></p>
<p>o de forma equivalente:</p>
<p><span class="math display">\[BC_{ij} = \frac{\sum_{k=1}^{n} |x_{ik} - x_{jk}|}{\sum_{k=1}^{n} (x_{ik} + x_{jk})}\]</span></p>
<p>donde: - <span class="math inline">\(BC_{ij}\)</span> = distancia de Bray-Curtis entre sitios <span class="math inline">\(i\)</span> y <span class="math inline">\(j\)</span> - <span class="math inline">\(C_{ij}\)</span> = suma de los valores menores para especies comunes entre ambos sitios - <span class="math inline">\(S_i\)</span>, <span class="math inline">\(S_j\)</span> = número total de individuos contados en los sitios <span class="math inline">\(i\)</span> y <span class="math inline">\(j\)</span> - <span class="math inline">\(x_{ik}\)</span>, <span class="math inline">\(x_{jk}\)</span> = abundancia de la especie <span class="math inline">\(k\)</span> en los sitios <span class="math inline">\(i\)</span> y <span class="math inline">\(j\)</span> - <span class="math inline">\(n\)</span> = número total de especies</p>
</section>
<section id="interpretación-2" class="level4">
<h4 class="anchored" data-anchor-id="interpretación-2">Interpretación:</h4>
<ul>
<li>Valores entre 0 y 1</li>
<li>0 = sitios idénticos en composición</li>
<li>1 = sitios completamente diferentes</li>
<li>No considera relaciones filogenéticas</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/bray_curtis.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
</section>
</section>
<section id="section-1" class="level2">
<h2 class="anchored" data-anchor-id="section-1"></h2>
<section id="distancia-de-jaccard" class="level3">
<h3 class="anchored" data-anchor-id="distancia-de-jaccard">Distancia de Jaccard</h3>
<p>La <strong>distancia de Jaccard</strong> mide la disimilitud entre dos conjuntos basándose únicamente en presencia/ausencia, sin considerar abundancias. Es útil para datos binarios.</p>
<section id="fórmula-del-índice-de-jaccard" class="level4">
<h4 class="anchored" data-anchor-id="fórmula-del-índice-de-jaccard">Fórmula del índice de Jaccard:</h4>
<p><span class="math display">\[J(A,B) = \frac{|A \cap B|}{|A \cup B|}\]</span></p>
</section>
<section id="fórmula-de-la-distancia-de-jaccard" class="level4">
<h4 class="anchored" data-anchor-id="fórmula-de-la-distancia-de-jaccard">Fórmula de la distancia de Jaccard:</h4>
<p><span class="math display">\[D_J(A,B) = 1 - J(A,B) = 1 - \frac{|A \cap B|}{|A \cup B|}\]</span></p>
<p>o en términos ecológicos:</p>
<p><span class="math display">\[J = \frac{c}{a + b - c}\]</span></p>
<p>donde: - <span class="math inline">\(J\)</span> = índice de similitud de Jaccard - <span class="math inline">\(a\)</span> = número de especies presentes en el sitio A - <span class="math inline">\(b\)</span> = número de especies presentes en el sitio B<br>
- <span class="math inline">\(c\)</span> = número de especies presentes en ambos sitios A y B - <span class="math inline">\(D_J\)</span> = distancia de Jaccard</p>
</section>
<section id="interpretación-3" class="level4">
<h4 class="anchored" data-anchor-id="interpretación-3">Interpretación:</h4>
<ul>
<li>Valores entre 0 y 1</li>
<li>0 = sitios idénticos (especies presentes son las mismas)</li>
<li>1 = sitios completamente diferentes (no comparten especies)</li>
<li>No considera abundancias, solo presencia/ausencia</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/distancia_jaccard.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
</section>
</section>
<section id="unifrac" class="level2">
<h2 class="anchored" data-anchor-id="unifrac">UniFrac</h2>
<p><strong>UniFrac</strong> es una métrica de distancia filogenética utilizada para comparar comunidades biológicas. Incorpora información sobre las relaciones filogenéticas entre los organismos observados, diferenciándose de medidas como Bray-Curtis.</p>
<section id="unifrac-no-ponderado-unweighted" class="level4">
<h4 class="anchored" data-anchor-id="unifrac-no-ponderado-unweighted">UniFrac No Ponderado (Unweighted):</h4>
<p><span class="math display">\[U(A,B) = \frac{\sum_{i=1}^{N} l_i \times |A_i - B_i|}{\sum_{i=1}^{N} l_i \times \max(A_i, B_i)}\]</span></p>
<p>donde: - <span class="math inline">\(U(A,B)\)</span> = distancia UniFrac no ponderada entre muestras A y B - <span class="math inline">\(N\)</span> = número de nodos en el árbol filogenético - <span class="math inline">\(l_i\)</span> = longitud de la rama entre el nodo <span class="math inline">\(i\)</span> y su nodo parental - <span class="math inline">\(A_i\)</span> = indicador (1 o 0) de presencia de la muestra A en las ramas descendientes del nodo <span class="math inline">\(i\)</span> - <span class="math inline">\(B_i\)</span> = indicador (1 o 0) de presencia de la muestra B en las ramas descendientes del nodo <span class="math inline">\(i\)</span></p>
</section>
<section id="unifrac-ponderado-weighted" class="level4">
<h4 class="anchored" data-anchor-id="unifrac-ponderado-weighted">UniFrac Ponderado (Weighted):</h4>
<p><span class="math display">\[W(A,B) = \sum_{i=1}^{N} l_i \times \left|\frac{A_i}{A_t} - \frac{B_i}{B_t}\right|\]</span></p>
<p>donde: - <span class="math inline">\(W(A,B)\)</span> = distancia UniFrac ponderada entre muestras A y B - <span class="math inline">\(A_i\)</span> = número de secuencias de la muestra A en las ramas descendientes del nodo <span class="math inline">\(i\)</span> - <span class="math inline">\(A_t\)</span> = número total de secuencias en la muestra A - <span class="math inline">\(B_i\)</span> = número de secuencias de la muestra B en las ramas descendientes del nodo <span class="math inline">\(i\)</span> - <span class="math inline">\(B_t\)</span> = número total de secuencias en la muestra B</p>
</section>
<section id="interpretación-4" class="level4">
<h4 class="anchored" data-anchor-id="interpretación-4">Interpretación:</h4>
<ul>
<li>Valores entre 0 y 1</li>
<li>0 = comunidades idénticas filogenéticamente</li>
<li>1 = comunidades completamente diferentes filogenéticamente</li>
</ul>
</section>
<section id="unweighted-unifrac-distance-permanova" class="level3">
<h3 class="anchored" data-anchor-id="unweighted-unifrac-distance-permanova">Unweighted UniFrac Distance: PERMANOVA</h3>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="tomato_16S_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="tomato_16S_files/figure-html/unnamed-chunk-24-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="gjvyybgokl" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#gjvyybgokl table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#gjvyybgokl thead, #gjvyybgokl tbody, #gjvyybgokl tfoot, #gjvyybgokl tr, #gjvyybgokl td, #gjvyybgokl th {
  border-style: none;
}

#gjvyybgokl p {
  margin: 0;
  padding: 0;
}

#gjvyybgokl .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#gjvyybgokl .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#gjvyybgokl .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#gjvyybgokl .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#gjvyybgokl .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gjvyybgokl .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gjvyybgokl .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gjvyybgokl .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#gjvyybgokl .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#gjvyybgokl .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#gjvyybgokl .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#gjvyybgokl .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#gjvyybgokl .gt_spanner_row {
  border-bottom-style: hidden;
}

#gjvyybgokl .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#gjvyybgokl .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#gjvyybgokl .gt_from_md > :first-child {
  margin-top: 0;
}

#gjvyybgokl .gt_from_md > :last-child {
  margin-bottom: 0;
}

#gjvyybgokl .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#gjvyybgokl .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#gjvyybgokl .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#gjvyybgokl .gt_row_group_first td {
  border-top-width: 2px;
}

#gjvyybgokl .gt_row_group_first th {
  border-top-width: 2px;
}

#gjvyybgokl .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gjvyybgokl .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#gjvyybgokl .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#gjvyybgokl .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gjvyybgokl .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gjvyybgokl .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#gjvyybgokl .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#gjvyybgokl .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#gjvyybgokl .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gjvyybgokl .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gjvyybgokl .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#gjvyybgokl .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gjvyybgokl .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#gjvyybgokl .gt_left {
  text-align: left;
}

#gjvyybgokl .gt_center {
  text-align: center;
}

#gjvyybgokl .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#gjvyybgokl .gt_font_normal {
  font-weight: normal;
}

#gjvyybgokl .gt_font_bold {
  font-weight: bold;
}

#gjvyybgokl .gt_font_italic {
  font-style: italic;
}

#gjvyybgokl .gt_super {
  font-size: 65%;
}

#gjvyybgokl .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#gjvyybgokl .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#gjvyybgokl .gt_indent_1 {
  text-indent: 5px;
}

#gjvyybgokl .gt_indent_2 {
  text-indent: 10px;
}

#gjvyybgokl .gt_indent_3 {
  text-indent: 15px;
}

#gjvyybgokl .gt_indent_4 {
  text-indent: 20px;
}

#gjvyybgokl .gt_indent_5 {
  text-indent: 25px;
}

#gjvyybgokl .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#gjvyybgokl div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="Group-1" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 1</th>
<th id="Group-2" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Group 2</th>
<th id="Sample-size" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Sample size</th>
<th id="Permutations" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Permutations</th>
<th id="pseudo-F" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">pseudo-F</th>
<th id="p-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">p-value</th>
<th id="q-value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">q-value</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Group 1">NC</td>
<td class="gt_row gt_left" headers="Group 2">Tomato</td>
<td class="gt_row gt_right" headers="Sample size">12</td>
<td class="gt_row gt_right" headers="Permutations">999</td>
<td class="gt_row gt_right" headers="pseudo-F">5.749228</td>
<td class="gt_row gt_right" headers="p-value">0.008</td>
<td class="gt_row gt_right" headers="q-value">0.008</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
</section>
</section>
<section id="taxonomía" class="level1">
<h1>Taxonomía</h1>
<section id="descarga-de-la-base-de-datos-silva" class="level3">
<h3 class="anchored" data-anchor-id="descarga-de-la-base-de-datos-silva">Descarga de la base de datos SILVA</h3>
<p>Descarga la base de datos SILVA en su versión 138.2, específicamente el conjunto de datos SSURef_NR99. Este conjunto contiene secuencias del gen 16S rRNA agrupadas al 99% de similitud, lo que reduce redundancia y facilita el procesamiento.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> rescript get-silva-data <span class="dt">\</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>--p-version <span class="st">'138.2'</span> <span class="dt">\</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>--p-target <span class="st">'SSURef_NR99'</span> <span class="dt">\</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>--o-silva-sequences silva-138.2-ssu-nr99-rna-seqs.qza <span class="dt">\</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>--o-silva-taxonomy silva-138.2-ssu-nr99-tax.qza</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="conversión-de-rna-a-dna" class="level3">
<h3 class="anchored" data-anchor-id="conversión-de-rna-a-dna">Conversión de RNA a DNA</h3>
<p>Convierte las secuencias de RNA descargadas a secuencias de DNA. Esto es necesario porque los análisis posteriores, incluyendo el entrenamiento del clasificador.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> rescript reverse-transcribe <span class="dt">\</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>--i-rna-sequences silva-138.2-ssu-nr99-rna-seqs.qza <span class="dt">\ </span>   <span class="at">--o-dna-sequences</span> silva-138.2-ssu-nr99-seqs.qza</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="eliminación-de-secuencias-de-baja-calidad" class="level3">
<h3 class="anchored" data-anchor-id="eliminación-de-secuencias-de-baja-calidad">Eliminación de secuencias de baja calidad</h3>
<p>Filtra y elimina secuencias de baja calidad o con posibles errores, como aquellas con bases ambiguas (ej. ‘N’) o secuencias con regiones repetitivas largas que podrían afectar la precisión del clasificador.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> rescript cull-seqs <span class="dt">\</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>--i-sequences silva-138.2-ssu-nr99-seqs.qza <span class="dt">\</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>--o-clean-sequences silva-138.2-ssu-nr99-seqs-cleaned.qza</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="filtrado-de-secuencias-por-longitud-y-taxón" class="level3">
<h3 class="anchored" data-anchor-id="filtrado-de-secuencias-por-longitud-y-taxón">Filtrado de secuencias por longitud y taxón</h3>
<p>Filtra las secuencias según su longitud mínima, dependiendo del grupo taxonómico:</p>
<ul>
<li><p>Archaea: ≥900 bp</p></li>
<li><p>Bacteria: ≥1200 bp</p></li>
<li><p>Eukaryota: ≥1400 bp</p></li>
</ul>
<p>Esto asegura que solo se utilicen secuencias completas o casi completas.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> rescript filter-seqs-length-by-taxon <span class="dt">\</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>--i-sequences silva-138.2-ssu-nr99-seqs-cleaned.qza <span class="dt">\</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>--i-taxonomy silva-138.2-ssu-nr99-tax.qza <span class="dt">\</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>--p-labels Archaea Bacteria Eukaryota <span class="dt">\</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>--p-min-lens 900 1200 1400 <span class="dt">\</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>--o-filtered-seqs silva-138.2-ssu-nr99-seqs-filt.qza <span class="dt">\ </span>   <span class="at">--o-discarded-seqs</span> silva-138.2-ssu-nr99-seqs-discard.qza</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="dereplicación-de-secuencias" class="level3">
<h3 class="anchored" data-anchor-id="dereplicación-de-secuencias">Dereplicación de secuencias</h3>
<p>Elimina secuencias duplicadas, manteniendo solo las únicas junto con su información taxonómica. Esto reduce la carga computacional sin perder diversidad taxonómica.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> rescript dereplicate <span class="dt">\</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>--i-sequences silva-138.2-ssu-nr99-seqs-filt.qza  <span class="dt">\</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>--i-taxa silva-138.2-ssu-nr99-tax.qza <span class="dt">\</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>--p-mode <span class="st">'uniq'</span> <span class="dt">\</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>--o-dereplicated-sequences silva-138.2-ssu-nr99-seqs-derep-uniq.qza <span class="dt">\</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>--o-dereplicated-taxa silva-138.2-ssu-nr99-tax-derep-uniq.qza</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="extracción-de-la-región-v4" class="level3">
<h3 class="anchored" data-anchor-id="extracción-de-la-región-v4">Extracción de la región V4</h3>
<p>Extrae la región hipervariable V4 del gen 16S rRNA usando los cebadores 515F (forward) y 806R (reverse). Esta región es comúnmente utilizada en estudios de microbiomas, por lo que el clasificador se optimiza para datos que la contengan.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier extract-reads <span class="dt">\</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>--i-sequences silva-138.2-ssu-nr99-seqs-derep-uniq.qza <span class="dt">\</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>--p-f-primer GTGYCAGCMGCCGCGGTAA <span class="dt">\</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>--p-r-primer GGACTACNVGGGTWTCTAAT <span class="dt">\</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>--p-n-jobs 5 <span class="dt">\</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>--p-read-orientation <span class="st">'forward'</span> <span class="dt">\</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>--o-reads silva-138.2-ssu-nr99-seqs-515f-806r.qza</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="dereplicación-de-la-región-v4" class="level3">
<h3 class="anchored" data-anchor-id="dereplicación-de-la-región-v4">Dereplicación de la región V4</h3>
<p>Realiza una segunda dereplicación, esta vez sobre las secuencias de la región V4, eliminando duplicados adicionales que puedan haber surgido tras la extracción.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="ex">iime</span> rescript dereplicate <span class="dt">\</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>--i-sequences silva-138.2-ssu-nr99-seqs-515f-806r.qza <span class="dt">\</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>--i-taxa silva-138.2-ssu-nr99-tax-derep-uniq.qza <span class="dt">\</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>--p-mode <span class="st">'uniq'</span> <span class="dt">\</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>--o-dereplicated-sequences silva-138.2-ssu-nr99-seqs-515f-806r-uniq.qza <span class="dt">\</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>--o-dereplicated-taxa  silva-138.2-ssu-nr99-tax-515f-806r-derep-uniq.qza</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="entrenamiento-del-clasificador-naive-bayes" class="level3">
<h3 class="anchored" data-anchor-id="entrenamiento-del-clasificador-naive-bayes">Entrenamiento del clasificador Naive Bayes</h3>
<p>Entrena un clasificador Naive Bayes utilizando las secuencias únicas de la región V4 y su taxonomía. Este clasificador puede usarse luego para asignar taxonomía a secuencias 16S de muestras de microbiomas.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier fit-classifier-naive-bayes <span class="dt">\</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>--i-reference-reads silva-138.2-ssu-nr99-seqs-515f-806r-uniq.qza <span class="dt">\</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>--i-reference-taxonomy silva-138.2-ssu-nr99-tax-515f-806r-derep-uniq.qza <span class="dt">\</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>--o-classifier silva-138.2-ssu-nr99-515f-806r-classifier.qza</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="diversidadabundancía" class="level2">
<h2 class="anchored" data-anchor-id="diversidadabundancía">Diversidad/abundancía</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier classify-sklearn <span class="dt">\</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-classifier</span> silva-138.2-ssu-nr99-515f-806r-classifier.qza <span class="dt">\</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-reads</span> rep-seqs-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-classification</span> taxonomy.qza</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> metadata tabulate <span class="dt">\</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--m-input-file</span> taxonomy.qza <span class="dt">\</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> taxonomy.qzv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> taxa barplot <span class="dt">\</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-table</span> table-dada2-tomato-16s.qza <span class="dt">\</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-taxonomy</span> taxonomy.qza <span class="dt">\</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> Experiment_Design.tsv <span class="dt">\</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> taxa-bar-plots.qzv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/rect469.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>